

<!doctype html>






<html  data-paige="Paige theme from https://github.com/willfaught/paige"   lang="en-us" >

    










    

    

    




<head>
    

    

<meta charset="utf-8">







<meta content="#0d6efd" name="msapplication-TileColor">


    <meta content="/browserconfig.xml" name="msapplication-config">


<meta content="https://github.com/willfaught/paige" name="theme">
<meta content="#0d6efd" name="theme-color">
<meta content="width=device-width, initial-scale=1" name="viewport">

<meta property="og:url" content="/workflows/">
  <meta property="og:title" content="Nanopore workflow">
  <meta property="og:description" content=" Table of Contents Molecular Procedures Worked Example Real-Time Example Molecular Procedures DNA Extraction Library Preparation Flow Cell Wash and Storage DNA Extraction Download DNA Extraction Protocol Library Preparation Download Nanopore Native Barcode Library Prep PDF Flow Cell Wash and Storage Download Flow Cell Wash and Storage Protocol PDF Worked Example The goal of this portion of the workshop is to practice the basic bioinformatic workflow required to process nanopore adaptive sampling sequence data. Specifically, we will map reads to reference sequences, filter out host sequences, and perform homology searches against known pathogen sequences. This workflow uses publicly available data published by Kipp et al. (2023).">
  <meta property="og:locale" content="en_us">
  <meta property="og:type" content="website">


  <meta name="twitter:card" content="summary">
  <meta name="twitter:title" content="Nanopore workflow">
  <meta name="twitter:description" content=" Table of Contents Molecular Procedures Worked Example Real-Time Example Molecular Procedures DNA Extraction Library Preparation Flow Cell Wash and Storage DNA Extraction Download DNA Extraction Protocol Library Preparation Download Nanopore Native Barcode Library Prep PDF Flow Cell Wash and Storage Download Flow Cell Wash and Storage Protocol PDF Worked Example The goal of this portion of the workshop is to practice the basic bioinformatic workflow required to process nanopore adaptive sampling sequence data. Specifically, we will map reads to reference sequences, filter out host sequences, and perform homology searches against known pathogen sequences. This workflow uses publicly available data published by Kipp et al. (2023).">



    
        <title>Nanopore workflow Â· Introduction</title>
    

    


    <link href="../apple-touch-icon.png" rel="apple-touch-icon" sizes="180x180">



    <link href="../favicon.ico" rel="shortcut icon">



    <link href="../favicon.png" rel="icon" type="image/png">



    <link href="../favicon.svg" rel="icon" type="image/svg+xml">



    <link href="../favicon-16x16.png" rel="icon" sizes="16x16" type="image/png">



    <link href="../favicon-32x32.png" rel="icon" sizes="32x32" type="image/png">



    <link color="#0d6efd" href="../safari-pinned-tab.svg" rel="mask-icon">



    <link href="../site.webmanifest" rel="manifest">
























    

    
        
        
        
    

    
        
    

    
    
    


<link  crossorigin="anonymous"  href="../css/paige/bootstrap/6186ad141d391e4c0c10d2e9cb4eaeaadf75e809-paige.min.90e8e43521c2ef40c030c3979ea72bfc4a04397b9995dce5eadbe91e9770a96b.css"  integrity="sha256-kOjkNSHC70DAMMOXnqcr/EoEOXuZldzl6tvpHpdwqWs="   referrerpolicy="no-referrer"  rel="stylesheet">



















    

    

    

    
    
    


<link  crossorigin="anonymous"  href="../css/paige/bootstrap-icons/bootstrap-icons.min.540116b7267d4b2410c74086ae6497566928c6d5715900be7ce7279ad45addcd.css"  integrity="sha256-VAEWtyZ9SyQQx0CGrmSXVmkoxtVxWQC&#43;fOcnmtRa3c0="   referrerpolicy="no-referrer"  rel="stylesheet">






    

<style>

    


.paige-figure-numbered {
    counter-increment: paige-figure-numbered;
}

.paige-figure-numbered > div > figure > figcaption::before {
    content: "Figure " counter(paige-figure-numbered) ": ";
}

.paige-figure-numbered > div > figure > figcaption:empty::before {
    content: "Figure " counter(paige-figure-numbered);
}

.paige-header-link {
    opacity: 0;
    margin-left: 0.5rem;
    position: absolute;
    transition: color 0.15s ease-in-out, opacity 0.15s ease-in-out;
}

.paige-header-link::after {
    content: "#";
}

.paige-quote .blockquote-footer {
    margin-top: 0;
}

.paige-quote blockquote {
    border-left: 0;
    border-right: 0;
    margin-bottom: 0;
    padding: 0;
}

blockquote {
    padding: 0.5rem 1rem;
    border-left: 0.25rem solid var(--bs-border-color);
    border-right: 0.25rem solid var(--bs-body-bg);
}

td,
th {
    padding: 0.25rem;
}

.highlight .chroma .hl,
.highlight .chroma .lnt {
    display: flex;
}

.paige-row-narrow:not(:last-child),
.paige-figure .paige-quote,
.paige-figure .paige-video,
.paige-figure .highlight pre.chroma,
.paige-figure .highlight .chroma pre,
.paige-figure .paige-quote blockquote,
.paige-figure figure > div > :last-child,
.paige-gallery .paige-figure,
.paige-gallery .paige-image,
blockquote > p:last-of-type {
    margin-bottom: 0;
}

.paige-figure,
.paige-gallery,
.paige-image,
.paige-quote,
.paige-row-narrow:last-child,
.paige-row-wide,
.paige-video,
#paige-content > *,
table {
    margin-bottom: 1rem;
}

.paige-header-link:focus,
.paige-header-link:hover,
:hover > .paige-header-link,
:target > .paige-header-link {
    opacity: 1;
}




    @media (prefers-color-scheme: dark) {
        /* Background */ .bg { color: #c9d1d9; background-color: var(--bs-body-bg); }
/* PreWrapper */ .chroma { color: #c9d1d9; background-color: var(--bs-body-bg); }
/* Other */ .chroma .x {  }
/* Error */ .chroma .err { color: #f85149 }
/* CodeLine */ .chroma .cl {  }
/* LineLink */ .chroma .lnlinks { outline: none; text-decoration: none; color: inherit }
/* LineTableTD */ .chroma .lntd { vertical-align: top; padding: 0; margin: 0; border: 0; }
/* LineTable */ .chroma .lntable { border-spacing: 0; padding: 0; margin: 0; border: 0; }
/* LineHighlight */ .chroma .hl { background-color: #4f4f4d }
/* LineNumbersTable */ .chroma .lnt { white-space: pre; user-select: none; margin-right: 0.4em; padding: 0 0.4em 0 0.4em;color: #64686c }
/* LineNumbers */ .chroma .ln { white-space: pre; user-select: none; margin-right: 0.4em; padding: 0 0.4em 0 0.4em;color: #6e7681 }
/* Line */ .chroma .line { display: flex; }
/* Keyword */ .chroma .k { color: #ff7b72 }
/* KeywordConstant */ .chroma .kc { color: #79c0ff }
/* KeywordDeclaration */ .chroma .kd { color: #ff7b72 }
/* KeywordNamespace */ .chroma .kn { color: #ff7b72 }
/* KeywordPseudo */ .chroma .kp { color: #79c0ff }
/* KeywordReserved */ .chroma .kr { color: #ff7b72 }
/* KeywordType */ .chroma .kt { color: #ff7b72 }
/* Name */ .chroma .n {  }
/* NameAttribute */ .chroma .na {  }
/* NameBuiltin */ .chroma .nb {  }
/* NameBuiltinPseudo */ .chroma .bp {  }
/* NameClass */ .chroma .nc { color: #f0883e; font-weight: bold }
/* NameConstant */ .chroma .no { color: #79c0ff; font-weight: bold }
/* NameDecorator */ .chroma .nd { color: #d2a8ff; font-weight: bold }
/* NameEntity */ .chroma .ni { color: #ffa657 }
/* NameException */ .chroma .ne { color: #f0883e; font-weight: bold }
/* NameFunction */ .chroma .nf { color: #d2a8ff; font-weight: bold }
/* NameFunctionMagic */ .chroma .fm {  }
/* NameLabel */ .chroma .nl { color: #79c0ff; font-weight: bold }
/* NameNamespace */ .chroma .nn { color: #ff7b72 }
/* NameOther */ .chroma .nx {  }
/* NameProperty */ .chroma .py { color: #79c0ff }
/* NameTag */ .chroma .nt { color: #7ee787 }
/* NameVariable */ .chroma .nv { color: #79c0ff }
/* NameVariableClass */ .chroma .vc {  }
/* NameVariableGlobal */ .chroma .vg {  }
/* NameVariableInstance */ .chroma .vi {  }
/* NameVariableMagic */ .chroma .vm {  }
/* Literal */ .chroma .l { color: #a5d6ff }
/* LiteralDate */ .chroma .ld { color: #79c0ff }
/* LiteralString */ .chroma .s { color: #a5d6ff }
/* LiteralStringAffix */ .chroma .sa { color: #79c0ff }
/* LiteralStringBacktick */ .chroma .sb { color: #a5d6ff }
/* LiteralStringChar */ .chroma .sc { color: #a5d6ff }
/* LiteralStringDelimiter */ .chroma .dl { color: #79c0ff }
/* LiteralStringDoc */ .chroma .sd { color: #a5d6ff }
/* LiteralStringDouble */ .chroma .s2 { color: #a5d6ff }
/* LiteralStringEscape */ .chroma .se { color: #79c0ff }
/* LiteralStringHeredoc */ .chroma .sh { color: #79c0ff }
/* LiteralStringInterpol */ .chroma .si { color: #a5d6ff }
/* LiteralStringOther */ .chroma .sx { color: #a5d6ff }
/* LiteralStringRegex */ .chroma .sr { color: #79c0ff }
/* LiteralStringSingle */ .chroma .s1 { color: #a5d6ff }
/* LiteralStringSymbol */ .chroma .ss { color: #a5d6ff }
/* LiteralNumber */ .chroma .m { color: #a5d6ff }
/* LiteralNumberBin */ .chroma .mb { color: #a5d6ff }
/* LiteralNumberFloat */ .chroma .mf { color: #a5d6ff }
/* LiteralNumberHex */ .chroma .mh { color: #a5d6ff }
/* LiteralNumberInteger */ .chroma .mi { color: #a5d6ff }
/* LiteralNumberIntegerLong */ .chroma .il { color: #a5d6ff }
/* LiteralNumberOct */ .chroma .mo { color: #a5d6ff }
/* Operator */ .chroma .o { color: #ff7b72; font-weight: bold }
/* OperatorWord */ .chroma .ow { color: #ff7b72; font-weight: bold }
/* Punctuation */ .chroma .p {  }
/* Comment */ .chroma .c { color: #8b949e; font-style: italic }
/* CommentHashbang */ .chroma .ch { color: #8b949e; font-style: italic }
/* CommentMultiline */ .chroma .cm { color: #8b949e; font-style: italic }
/* CommentSingle */ .chroma .c1 { color: #8b949e; font-style: italic }
/* CommentSpecial */ .chroma .cs { color: #8b949e; font-weight: bold; font-style: italic }
/* CommentPreproc */ .chroma .cp { color: #8b949e; font-weight: bold; font-style: italic }
/* CommentPreprocFile */ .chroma .cpf { color: #8b949e; font-weight: bold; font-style: italic }
/* Generic */ .chroma .g {  }
/* GenericDeleted */ .chroma .gd { color: #ffa198; background-color: #490202 }
/* GenericEmph */ .chroma .ge { font-style: italic }
/* GenericError */ .chroma .gr { color: #ffa198 }
/* GenericHeading */ .chroma .gh { color: #79c0ff; font-weight: bold }
/* GenericInserted */ .chroma .gi { color: #56d364; background-color: #0f5323 }
/* GenericOutput */ .chroma .go { color: #8b949e }
/* GenericPrompt */ .chroma .gp { color: #8b949e }
/* GenericStrong */ .chroma .gs { font-weight: bold }
/* GenericSubheading */ .chroma .gu { color: #79c0ff }
/* GenericTraceback */ .chroma .gt { color: #ff7b72 }
/* GenericUnderline */ .chroma .gl { text-decoration: underline }
/* TextWhitespace */ .chroma .w { color: #6e7681 }

    }

    @media (prefers-color-scheme: light) {
        /* Background */ .bg { background-color: #ffffff; }
/* PreWrapper */ .chroma { background-color: #ffffff; }
/* Other */ .chroma .x {  }
/* Error */ .chroma .err { color: #a61717; background-color: #e3d2d2 }
/* CodeLine */ .chroma .cl {  }
/* LineLink */ .chroma .lnlinks { outline: none; text-decoration: none; color: inherit }
/* LineTableTD */ .chroma .lntd { vertical-align: top; padding: 0; margin: 0; border: 0; }
/* LineTable */ .chroma .lntable { border-spacing: 0; padding: 0; margin: 0; border: 0; }
/* LineHighlight */ .chroma .hl { background-color: #ffffcc }
/* LineNumbersTable */ .chroma .lnt { white-space: pre; user-select: none; margin-right: 0.4em; padding: 0 0.4em 0 0.4em;color: #7f7f7f }
/* LineNumbers */ .chroma .ln { white-space: pre; user-select: none; margin-right: 0.4em; padding: 0 0.4em 0 0.4em;color: #7f7f7f }
/* Line */ .chroma .line { display: flex; }
/* Keyword */ .chroma .k { color: #000000; font-weight: bold }
/* KeywordConstant */ .chroma .kc { color: #000000; font-weight: bold }
/* KeywordDeclaration */ .chroma .kd { color: #000000; font-weight: bold }
/* KeywordNamespace */ .chroma .kn { color: #000000; font-weight: bold }
/* KeywordPseudo */ .chroma .kp { color: #000000; font-weight: bold }
/* KeywordReserved */ .chroma .kr { color: #000000; font-weight: bold }
/* KeywordType */ .chroma .kt { color: #445588; font-weight: bold }
/* Name */ .chroma .n {  }
/* NameAttribute */ .chroma .na { color: #008080 }
/* NameBuiltin */ .chroma .nb { color: #0086b3 }
/* NameBuiltinPseudo */ .chroma .bp { color: #999999 }
/* NameClass */ .chroma .nc { color: #445588; font-weight: bold }
/* NameConstant */ .chroma .no { color: #008080 }
/* NameDecorator */ .chroma .nd { color: #3c5d5d; font-weight: bold }
/* NameEntity */ .chroma .ni { color: #800080 }
/* NameException */ .chroma .ne { color: #990000; font-weight: bold }
/* NameFunction */ .chroma .nf { color: #990000; font-weight: bold }
/* NameFunctionMagic */ .chroma .fm {  }
/* NameLabel */ .chroma .nl { color: #990000; font-weight: bold }
/* NameNamespace */ .chroma .nn { color: #555555 }
/* NameOther */ .chroma .nx {  }
/* NameProperty */ .chroma .py {  }
/* NameTag */ .chroma .nt { color: #000080 }
/* NameVariable */ .chroma .nv { color: #008080 }
/* NameVariableClass */ .chroma .vc { color: #008080 }
/* NameVariableGlobal */ .chroma .vg { color: #008080 }
/* NameVariableInstance */ .chroma .vi { color: #008080 }
/* NameVariableMagic */ .chroma .vm {  }
/* Literal */ .chroma .l {  }
/* LiteralDate */ .chroma .ld {  }
/* LiteralString */ .chroma .s { color: #dd1144 }
/* LiteralStringAffix */ .chroma .sa { color: #dd1144 }
/* LiteralStringBacktick */ .chroma .sb { color: #dd1144 }
/* LiteralStringChar */ .chroma .sc { color: #dd1144 }
/* LiteralStringDelimiter */ .chroma .dl { color: #dd1144 }
/* LiteralStringDoc */ .chroma .sd { color: #dd1144 }
/* LiteralStringDouble */ .chroma .s2 { color: #dd1144 }
/* LiteralStringEscape */ .chroma .se { color: #dd1144 }
/* LiteralStringHeredoc */ .chroma .sh { color: #dd1144 }
/* LiteralStringInterpol */ .chroma .si { color: #dd1144 }
/* LiteralStringOther */ .chroma .sx { color: #dd1144 }
/* LiteralStringRegex */ .chroma .sr { color: #009926 }
/* LiteralStringSingle */ .chroma .s1 { color: #dd1144 }
/* LiteralStringSymbol */ .chroma .ss { color: #990073 }
/* LiteralNumber */ .chroma .m { color: #009999 }
/* LiteralNumberBin */ .chroma .mb { color: #009999 }
/* LiteralNumberFloat */ .chroma .mf { color: #009999 }
/* LiteralNumberHex */ .chroma .mh { color: #009999 }
/* LiteralNumberInteger */ .chroma .mi { color: #009999 }
/* LiteralNumberIntegerLong */ .chroma .il { color: #009999 }
/* LiteralNumberOct */ .chroma .mo { color: #009999 }
/* Operator */ .chroma .o { color: #000000; font-weight: bold }
/* OperatorWord */ .chroma .ow { color: #000000; font-weight: bold }
/* Punctuation */ .chroma .p {  }
/* Comment */ .chroma .c { color: #999988; font-style: italic }
/* CommentHashbang */ .chroma .ch { color: #999988; font-style: italic }
/* CommentMultiline */ .chroma .cm { color: #999988; font-style: italic }
/* CommentSingle */ .chroma .c1 { color: #999988; font-style: italic }
/* CommentSpecial */ .chroma .cs { color: #999999; font-weight: bold; font-style: italic }
/* CommentPreproc */ .chroma .cp { color: #999999; font-weight: bold; font-style: italic }
/* CommentPreprocFile */ .chroma .cpf { color: #999999; font-weight: bold; font-style: italic }
/* Generic */ .chroma .g {  }
/* GenericDeleted */ .chroma .gd { color: #000000; background-color: #ffdddd }
/* GenericEmph */ .chroma .ge { color: #000000; font-style: italic }
/* GenericError */ .chroma .gr { color: #aa0000 }
/* GenericHeading */ .chroma .gh { color: #999999 }
/* GenericInserted */ .chroma .gi { color: #000000; background-color: #ddffdd }
/* GenericOutput */ .chroma .go { color: #888888 }
/* GenericPrompt */ .chroma .gp { color: #555555 }
/* GenericStrong */ .chroma .gs { font-weight: bold }
/* GenericSubheading */ .chroma .gu { color: #aaaaaa }
/* GenericTraceback */ .chroma .gt { color: #aa0000 }
/* GenericUnderline */ .chroma .gl { text-decoration: underline }
/* TextWhitespace */ .chroma .w { color: #bbbbbb }

    }


@media (prefers-reduced-motion: reduce) {
    .paige-header-link {
        transition: none;
    }
}




</style>


    
</head>

    <body>

        

        <div class="container">
            <div class="row">
                <div class="col mt-3" id="paige-site">
                    

                    














    





    



    <header id="paige-site-header">
        

        
            <div class="display-1 fw-bold  mb-2  paige-row-wide text-center" id="paige-site-title">Field Genomics Workshop</div>
        

        
            <div class="lead paige-row-wide text-center" id="paige-site-description">Ecuador 2024</div>
        

        
            
                <nav aria-label="Menu" class="paige-row-wide" id="paige-menu">
                    <ul class="align-items-center justify-content-center nav ">
                        
                            
                            
                            
                            
                            

                            <li class="nav-item ">
                                <a   class="  nav-link   text-decoration-underline "  href="../"    >Home</a>

                                
                            </li>
                        
                            
                            
                            
                            
                            

                            <li class="nav-item ">
                                <a   class="  nav-link   text-decoration-underline "  href="../getting-ready/"    >Getting Ready</a>

                                
                            </li>
                        
                            
                            
                            
                            
                            

                            <li class="nav-item ">
                                <a  aria-current="page"   class=" active  fw-bold text-body-emphasis    nav-link   text-decoration-underline "  href="../workflows/"    >Workflows</a>

                                
                            </li>
                        
                            
                            
                            
                            
                            

                            <li class="nav-item ">
                                <a   class="  nav-link   text-decoration-underline "  href="../slides/"    >Slides</a>

                                
                            </li>
                        
                            
                            
                            
                            
                            

                            <li class="nav-item ">
                                <a   class="  nav-link   text-decoration-underline "  href="../replicate-this-workshop/"    >Replicate This Workshop</a>

                                
                            </li>
                        
                            
                            
                            
                            
                            

                            <li class="nav-item ">
                                <a   class="  nav-link   text-decoration-underline "  href="../contactus/"    >Contact Us</a>

                                
                            </li>
                        
                    </ul>
                </nav>
            
        

        

        
    </header>


                    




















    



    
        
    




<article class="paige-kind-section paige-published" id="paige-page">
    <div class="align-items-center d-flex flex-column">
        




















    <header class="mw-100" id="paige-page-header">
        <div class="align-items-center d-flex flex-column">
            

            
                <h1 class="fw-bold" id="paige-page-title">Nanopore workflow</h1>
            

            

            

            

            

            
        </div>
    </header>



        
            <main class="mw-100" id="paige-content"></div>
<hr>
<h2 id="table-of-contents">Table of Contents<a aria-label="Link to this section" class="paige-header-link" href="#table-of-contents"></a></h2>
<ul>
<li><a href="#molecular-procedures">Molecular Procedures</a></li>
<li><a href="#worked-example">Worked Example</a></li>
<li><a href="#real-time-example">Real-Time Example</a></li>
</ul>
<br>
<h2 id="molecular-procedures">Molecular Procedures<a aria-label="Link to this section" class="paige-header-link" href="#molecular-procedures"></a></h2>
<!-- Buttons to toggle sections -->
<div style="text-align: center; margin-bottom: 20px;">
  <button onclick="toggleSection('dnaExtraction')"> DNA Extraction</button>
  <button onclick="toggleSection('libraryPrep')">Library Preparation</button>
  <button onclick="toggleSection('flowCell')"> Flow Cell Wash and Storage</button>
</div>
<!-- Containers for the PDFs -->
<div style="display: flex; flex-wrap: wrap; gap: 20px; justify-content: space-evenly;">
  <!-- DNA Extraction -->
  <div id="dnaExtraction" class="pdf-section" style="display: none; border: 1px solid #ccc; border-radius: 8px; width: 600px; padding: 15px; box-shadow: 2px 2px 8px rgba(0,0,0,0.1); text-align: center;">
    <h3>DNA Extraction</h3>
    <a href="../procedures/extraction.pdf" target="_blank">Download DNA Extraction Protocol</a>
    <iframe src="../procedures/extraction.pdf" width="100%" height="400" style="border: none; margin-top: 10px;"></iframe>
  </div>
  <!-- Library Preparation -->
  <div id="libraryPrep" class="pdf-section" style="display: none; border: 1px solid #ccc; border-radius: 8px; width: 600px; padding: 15px; box-shadow: 2px 2px 8px rgba(0,0,0,0.1); text-align: center;">
    <h3>Library Preparation</h3>
    <a href="../procedures/lib_prep.pdf" target="_blank">Download Nanopore Native Barcode Library Prep PDF</a>
    <iframe src="../procedures/lib_prep.pdf" width="100%" height="400" style="border: none; margin-top: 10px;"></iframe>
  </div>
  <!-- Flow Cell Wash and Storage -->
  <div id="flowCell" class="pdf-section" style="display: none; border: 1px solid #ccc; border-radius: 8px; width: 600px; padding: 15px; box-shadow: 2px 2px 8px rgba(0,0,0,0.1); text-align: center;">
    <h3>Flow Cell Wash and Storage</h3>
    <a href="../procedures/flow_cell_wash_kit.pdf" target="_blank">Download Flow Cell Wash and Storage Protocol PDF</a>
    <iframe src="../procedures/flow_cell_wash_kit.pdf" width="100%" height="400" style="border: none; margin-top: 10px;"></iframe>
  </div>
</div>
<!-- JavaScript for toggling sections -->
<script>
  function toggleSection(sectionId) {
    var section = document.getElementById(sectionId);
    if (section.style.display === "none" || section.style.display === "") {
      section.style.display = "block";
      section.style.opacity = 0;
      setTimeout(function() { section.style.opacity = 1; }, 10); // Fade in
    } else {
      section.style.opacity = 0;
      setTimeout(function() { section.style.display = "none"; }, 500); // Fade out
    }
  }
</script>
<!-- CSS for fade effect -->
<style>
  .pdf-section {
    transition: opacity 0.5s ease;
  }
</style>
<style>
pre code {
    background-color: #2d2d2d; /* Dark gray background for the code block */
    border: 1px solid #555555; /* Light gray border around the code block */
    border-radius: 5px; /* Rounded corners for the box */
    padding: 10px; /* Adds some space inside the box */
    display: block; /* Ensures it's displayed as a block element */
    white-space: pre-wrap; /* Ensures the code wraps properly if it's too long */
    color: #ffffff; /* White text color for better contrast */
}
</style>
<br>
<hr>
<h1 id="worked-example">Worked Example<a aria-label="Link to this section" class="paige-header-link" href="#worked-example"></a></h1>
<p>The goal of this portion of the workshop is to practice the basic bioinformatic workflow required to process nanopore adaptive sampling sequence data. Specifically, we will map reads to reference sequences, filter out host sequences, and perform homology searches against known pathogen sequences. This workflow uses publicly available data published by <a href="https://www.nature.com/articles/s41598-023-37134-9">Kipp et al. (2023)</a>.</p>
<p><a href="https://www.nature.com/articles/s41598-023-37134-9">Kipp et al. (2023)</a> shows how NAS can be used to selectively sequence DNA from multiple bacterial tick-borne pathogens circulating in wild populations of the black-legged tick vector, <i>Ixodes scapularis</i>. They test the consistency of pathogen recovery using NAS enrichment and depletion methods, compared to more traditional Illumina metagenomic sequencing methods. NAS enrichment and depletion methods successfully recovered 4 bacterial tick-borne pathogens (<i>Borellia burgdorferi</i>; <i>Borella miyamotoi</i>; <i>Anaplasma phagocytophilum</i>; and <i>Eherlichia muris</i>) from 8 female <i>I. scapularis</i> ticks, and even resulted in the sequencing of the entire bacterial pathogen genomes. Importantly, their NAS results were consistent with Illumina 16S microbiome data.</p>
<p>For this exercise, we are only working with their depletion dataset, as that workflow is best aligned with the sequencing approach of this workshop.
The goal of this process is to use NAS for real-time metagenomic surveillance, targeting pathogen detection in field-collected samples. This approach allows for selective sequencing of desired genomic regions. By performing these steps, we can efficiently map sequencing reads to reference genomes, filter out host sequences, and perform homology searches against known pathogen databases. This workflow mirrors the methodology described in <a href="https://www.nature.com/articles/s41598-023-37134-9">Kipp et al. (2023)</a>, where nanopore sequencing was used to study tick-borne bacterial pathogens.</p>
<p>Before starting with Step 1: (1) install the <a href="../programs">Required Software</a>.   (2) make sure you have downloaded Kipp&rsquo;s data as instructed in <a href="../getting-ready">Getting Ready</a>.</p>
<br>
<h2 id="step-1-getting-started">Step 1: Getting Started<a aria-label="Link to this section" class="paige-header-link" href="#step-1-getting-started"></a></h2>
<p>In this step, we are using basic Linux commands to establish our file/folder structure. Mac and Linux users may open a terminal. Windows users should open their Ubuntu terminal via WSL.</p>
<p>Print your working directory (i.e., determine where your terminal window is located within the file structure of your computer):
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="nb">pwd</span>  <span class="c1"># type this command and hit enter</span></span></span></code></pre></div></p>
<p>Home directory for Windows users via Ubuntu should be:</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="se">\\</span>wsl.localhost<span class="se">\U</span>buntu<span class="se">\h</span>ome<span class="se">\&lt;</span>USERNAME&gt;<span class="se">\f</span>ield_genomics<span class="se">\d</span>ata_example  <span class="c1"># &lt;USERNAME&gt; was defined when installing Ubuntu</span>
</span></span></code></pre></div><p>Make a directory
(folder) for this workflow:
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-python" data-lang="python"><span class="line"><span class="cl"><span class="n">mkdir</span> <span class="n">field_genomics</span></span></span></code></pre></div></p>
<p>Navigate into that directory:
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="nb">cd</span> ~/field_genomics/</span></span></code></pre></div></p>
<p>Create a new subdirectory to store example data:
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">mkdir data_example</span></span></code></pre></div></p>
<p>Create another subdirectory to store results:
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">mkdir output_example</span></span></code></pre></div></p>
<p>Create a third subdirectory to store a BLAST database:
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">mkdir blast_database</span></span></code></pre></div></p>
<p>Create a fourth subdirectory to summarize results from the example:
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">mkdir summary_example</span></span></code></pre></div></p>
<p>Copy or move the 3 data files from <a href="../getting-ready">Getting Ready</a> into the data_example/ directory.
This can be done in a File Manager or Finder window via drag-and-drop or via the command line.
Ensure that the files downloaded are named as follows, as sometimes downloading/extracting zip files from Google Drive can change file names:</p>
<ul>
<li>
<p><code>Kipp_etal_depletion_reference.fasta</code></p>
</li>
<li>
<p><code>SRR17281099.fasta.gz</code></p>
</li>
<li>
<p><code>Kipp_etal_enrichment_reference.fasta</code></p>
<p>Example code for transferring these files from your <code>Downloads/</code> folder to the <code>data_example/</code> folder:</p>
</li>
</ul>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="nb">cd</span> Downloads/
</span></span><span class="line"><span class="cl">mv SRR17281099.fasta.gz /&lt;YOUR_PATH&gt;/field_genomics/
</span></span><span class="line"><span class="cl">mv Kipp_etal_* &lt;YOUR_PATH&gt;/field_genomics/
</span></span></code></pre></div><p><strong>Note</strong>: For Windows users, ensure that your computer does not suspend while running analyses. You need to repace <code>&lt;YOUR_PATH&gt;</code> with the output of your <code>pwd</code> command from above.</p>
<br>
<h2 id="step-2-burrows-wheeler-aligner-bwa">Step 2: Burrows-Wheeler Aligner (BWA)<a aria-label="Link to this section" class="paige-header-link" href="#step-2-burrows-wheeler-aligner-bwa"></a></h2>
<p><a href="https://github.com/lh3/bwa">Burrows-Wheeler Alignment</a> (BWA) is a software package that is used to align reads to a reference genome. First, we must index the reference genome. Indexing a reference genome is a process that speeds up sequence alignment by creating a data structure, allowing rapid matching of DNA sequences (reads) to the reference genome. This is how to index a reference genome:</p>
<h3 id="variables">Variables:<a aria-label="Link to this section" class="paige-header-link" href="#variables"></a></h3>
<ul>
<li><strong>Reference Genome Path</strong>: <code>data_example/Kipp_etal_depletion_reference.fasta</code></li>
</ul>
<h3 id="code">Code:<a aria-label="Link to this section" class="paige-header-link" href="#code"></a></h3>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">bwa index data_example/Kipp_etal_depletion_reference.fasta</span></span></code></pre></div>
<br>
<ol>
<li>
<p><strong>Input: Reference Genome</strong></p>
<ul>
<li>A FASTA file containing the reference genome sequences.</li>
<li>Example:
<pre tabindex="0"><code>&gt;chr1  
AGCTTAGCTAGCTACGATCGATCG  
&gt;chr2  
GCTAGCTAGCTAGCTGATCGTACG  
</code></pre></li>
</ul>
</li>
<li>
<p><strong>Indexing with BWA</strong></p>
<ul>
<li>BWA generates auxiliary files (<code>.amb</code>,<code>.ann</code>,<code>.bwt</code>,<code>.pac</code>,<code>.sa</code>) such as suffix arrays or FM-indexes, that make the search process more efficient.</li>
<li>These files allow the alignment tool to quickly locate where short DNA sequences (reads) map to the genome.</li>
</ul>
</li>
</ol>
<br>
<h2 id="step-3-align-nanopore-reads-to-a-reference-genome">Step 3: Align Nanopore Reads to a Reference Genome<a aria-label="Link to this section" class="paige-header-link" href="#step-3-align-nanopore-reads-to-a-reference-genome"></a></h2>
<p><strong>Description</strong>:
This step aligns the Nanopore sequencing reads to the reference genome. BWA-MEM is the algorithm in BWA that is best for aligning long reads (70bp or more).  <a href="https://samtools.github.io/hts-specs/SAMv1.pdf">SAM</a> files (the ouput file format of this step) are a formatted, plain-text file that contains aligment information for sequences mapped to a reference. SAM files can also contain unmapped sequences.</p>
<div style="background-color: #f8f9fa; padding: 15px; border-radius: 8px; text-align: center;"> <img src="../procedures/mapping.png" alt="Alignment Process" style="max-width: 100%; height: auto;"/> </div>
<p>Photo credit: <a href="https://usegalaxy.org/">Galaxy</a></p>
<h3 id="variables-1">Variables:<a aria-label="Link to this section" class="paige-header-link" href="#variables-1"></a></h3>
<ul>
<li><strong>Reference Genome Path</strong>: <code>data_example/Kipp_etal_depletion_reference.fasta</code></li>
<li><strong>Nanopore Sequences Path</strong>: <code>data_example/SRR17281099.fasta.gz</code></li>
<li><strong>Output SAM File Path</strong>: <code>output_example/Kipp_etal_alig_depletion.sam</code></li>
</ul>
<h3 id="code-1">Code:<a aria-label="Link to this section" class="paige-header-link" href="#code-1"></a></h3>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1">#template command for nanopore reads: bwa mem -x ont2d &lt;reference_genome.fasta&gt; &lt;raw_reads.fasta.gz&gt; &lt;output.sam&gt;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">bwa mem -x ont2d data_example/Kipp_etal_depletion_reference.fasta data_example/SRR17281099.fasta.gz &gt; output_example/Kipp_etal_alig_depletion.sam
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># -x is a flag to specify specialized alignment stratagies, in this case, ont2d = Oxford Nanopore Reads</span></span></span></code></pre></div>
<br>
<h2 id="step-4-remove-host-sequence-data">Step 4: Remove Host Sequence Data<a aria-label="Link to this section" class="paige-header-link" href="#step-4-remove-host-sequence-data"></a></h2>
<p><strong>Description</strong>:
Here, we remove reads that mapped to the host genome, retaining only unmapped sequences in a separate fasta file. These sequences represent reads that are NOT the host (e.g., bacterial DNA, fungal DNA, contamination, etc.).</p>
<h3 id="variables-2">Variables:<a aria-label="Link to this section" class="paige-header-link" href="#variables-2"></a></h3>
<ul>
<li><strong>Input SAM File Path</strong>: <code>output_example/Kipp_etal_alig_depletion.sam</code></li>
<li><strong>Output Unmapped Reads Path</strong>: <code>output_example/Kipp_etal_unmapped_depletion.fasta</code></li>
</ul>
<h3 id="code-2">Code:<a aria-label="Link to this section" class="paige-header-link" href="#code-2"></a></h3>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">samtools fasta -f <span class="m">4</span> output_example/Kipp_etal_alig_depletion.sam &gt; output_example/Kipp_etal_unmapped_depletion.fasta
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#samtools -f is a flag that means &#34;flag.&#34; In this case, we specify 4, which translates to filter that fasta file to keep only unmapped sequences</span></span></span></code></pre></div>
<br>
<p><strong>Description</strong>:
<a href="https://www.htslib.org/">samtools</a> is a suite of programs used for interacting with high-throughput sequencing data. It consists of samtools, BCFtools, and HTSlib.</p>
<br>
<h2 id="step-5-basic-local-alignment-search-tool-blast">Step 5: Basic Local Alignment Search Tool (BLAST)<a aria-label="Link to this section" class="paige-header-link" href="#step-5-basic-local-alignment-search-tool-blast"></a></h2>
<p><strong>Description</strong>:
Next, we want to build a local database of tick-borne pathogen sequences. We will then compare our unaligned Nanopore reads to that database of reference pathogen sequences and use the percent identity (or percent similarity) to determine which pathogens are present in the Nanopore sequence data. We will use National Center for Biotechnology Information&rsquo;s (NCBI) BLAST (Basic Local Alignment Search Tool) tool to both (1) build the local database of know pathogens and (2) compare our data to it. The pathogen sequences used to build the reference database can be mined from GenBank/SRA, DDBJ, EuPathDB, or other sequence databases. Here, we build a database from the enrichment fasta file from Kipp et al. (2023) that contains 52 partial or complete genomes of tick-borne pathogens (see their <a href="https://static-content.springer.com/esm/art%3A10.1038%2Fs41598-023-37134-9/MediaObjects/41598_2023_37134_MOESM1_ESM.pdf">Supplementary Table 1</a>).</p>
<h3 id="code-3">Code:<a aria-label="Link to this section" class="paige-header-link" href="#code-3"></a></h3>
<h4 id="create-blast-database">Create BLAST Database:<a aria-label="Link to this section" class="paige-header-link" href="#create-blast-database"></a></h4>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">makeblastdb -in data_example/Kipp_etal_enrichment_reference.fasta -parse_seqids -blastdb_version <span class="m">5</span> -title <span class="s2">&#34;Kipp_etal_enrichment&#34;</span> -dbtype nucl -out blast_database/Kipp_etal_enrichment_reference
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#-blastdb_version 5 = Version of blastdb you would like. Version 5 (taxonomy aware) is the default starting with the 2.10.0 release. Value must be 4 or 5.</span>
</span></span><span class="line"><span class="cl"><span class="c1"># -title &#34;Kipp etal_enrichment&#34; = 	Title for BLAST database. If not set, the input file name will be used</span>
</span></span><span class="line"><span class="cl"><span class="c1"># -dbtype nucl = Molecule type of input, values can be nucl or prot.</span></span></span></code></pre></div>
<br>
<p><strong>Description</strong>:
The above line of code creates a local BLAST database from the sequences in Kipp et al.&rsquo;s (2023) enrichment fasta file. This database will be used as a reference against which we can compare our Nanopore sequence data to determine what pathogens are present in the sample.</p>
<br>
<h4 id="run-blast-query">Run BLAST Query:<a aria-label="Link to this section" class="paige-header-link" href="#run-blast-query"></a></h4>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">blastn -query output_example/Kipp_etal_unmapped_depletion.fasta -db blast_database/Kipp_etal_enrichment_reference -out output_example/Kipp_etal_blast_results.txt</span></span></code></pre></div>
<br>
<p><strong>Description</strong>:
This step uses the BLAST algorithm to search for each unmapped Nanopore sequence in the local pathogen reference database, and produce a text file with subject-query matches. In this context, the <code>subject</code> is the reference pathogen sequence and the query is the unmapped Nanopore read. A match occurs when there is &gt;70% identity between the subject and query.
<br></p>
<p>To summarize the BLAST results, run another BLAST query defining a tab-delimited output: -outfmt 6.</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">blastn -query output_example/Kipp_etal_unmapped_depletion.fasta -db blast_database/Kipp_etal_enrichment_reference -out output_example/Kipp_etal_blast_results_table.txt -outfmt <span class="m">6</span>
</span></span><span class="line"><span class="cl">	<span class="c1">#Options supplied to -outfmt (out format) can be customized to include alignment length,      percent identity, etc.</span>
</span></span></code></pre></div><p>BLASTn tabular output format 6 guide:</p>
<img src="../results/blasttab.jpg" alt="Alignment Process" style="max-width: 100%; height: auto; display: block; margin: 0 auto;"/>
<br> 
<h2 id="step-6-summarize-the-results">Step 6: Summarize the Results<a aria-label="Link to this section" class="paige-header-link" href="#step-6-summarize-the-results"></a></h2>
<p>Here, we can take a look at what a small part of our default BLAST query output looks like:</p>
<img src="../results/blast4.jpg" alt="Alignment Process" style="max-width: 100%; height: auto; display: block; margin: 0 auto;"/>
<br>
<p><strong>Lambda, K, H (Ungapped and Gapped)</strong></p>
<ul>
<li>These are statistical parameters used in BLAST to model the alignment scoring system.</li>
<li><strong>Lambda</strong>: Describes the expected distribution of alignment scores.</li>
<li><strong>K</strong>: Indicates the scale of the search space size.</li>
<li><strong>H</strong>: Describes the entropy or complexity of the scoring system.</li>
</ul>
<p><strong>Effective Search Space Used</strong></p>
<ul>
<li>The size of the search space after applying any filters or adjustments based on the query and database.</li>
</ul>
<p><strong>Query</strong></p>
<ul>
<li>The identifier of the input sequence, for example: <code>SRR17281099.315</code>.</li>
<li><strong>Length=564</strong>: Refers to the length of the query sequence (in nucleotides or amino acids, depending on the type of BLAST search).</li>
</ul>
<p><strong>Sequences Producing Significant Alignments</strong></p>
<ul>
<li>Sequences in the database that have significant matches with the query sequence.</li>
<li><strong>Example</strong>: <code>NC_021880.1_An_ph_str_JM</code></li>
<li><strong>Length=1481598</strong>: Indicates the length of the subject sequence (in nucleotides or amino acids).</li>
</ul>
<p><strong>Score</strong></p>
<ul>
<li>The alignment score in <strong>bits</strong>. Higher scores indicate better alignments.
Score = <strong>821 bits</strong>.</li>
</ul>
<p><strong>E-value (Expect Value)</strong></p>
<ul>
<li>The <strong>E-value</strong> measures statistical significance; smaller values indicate more significant alignments.</li>
</ul>
<p><strong>Identities</strong></p>
<ul>
<li>Refers to the number and percentage of identical matches between the query and subject.
<strong>532/570 (93%)</strong>, meaning that 532 out of 570 positions are identical.</li>
</ul>
<p><strong>Gaps</strong></p>
<ul>
<li>The number of gaps (insertions or deletions) in the alignment.
<strong>23/570 (4%)</strong>, meaning 23 positions out of 570 have gaps.</li>
</ul>
<p><strong>Strand</strong></p>
<ul>
<li>Indicates whether the alignment is on the same strand (plus/plus) or opposite strands (plus/minus).</li>
</ul>
<p><strong>Alignment</strong></p>
<ul>
<li>The actual sequence alignment between the query and subject:
<ul>
<li>Vertical bars (<code>|</code>) indicate identical matches.</li>
<li>A gap ( ) indicates an insertion or deletion.</li>
</ul>
</li>
</ul>
<hr>
<p><strong>Description</strong>:
Filter matches output from BLAST to retain only the query sequences that matched a reference sequence with &gt;97% identity.</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">awk -F<span class="s1">&#39;\t&#39;</span> <span class="s1">&#39;$3 &gt; 97&#39;</span> output_example/Kipp_etal_blast_results_table.txt &gt; summary_example/Kipp_etal_high_hits.txt
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># awk is a text processing command that allows you to filter, extract, and manipulate data from structured text, such as by selecting specific fields from a file based on a delimiter, applying conditions, and performing actions on matching lines</span>
</span></span><span class="line"><span class="cl"><span class="c1"># -F&#39;\t&#39; specifies tabs (\t) as the white-space character delimiting the input file</span>
</span></span><span class="line"><span class="cl"><span class="c1"># Only rows where the value in the third field ($3) is greater than 97 are written to the output file</span>
</span></span></code></pre></div><div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">cut -f1 summary_example/Kipp_etal_high_hits.txt <span class="p">|</span> sort <span class="p">|</span> uniq &gt; summary_example/Kipp_etal_high_hit_headers.txt
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">cut -f2 summary_example/Kipp_etal_high_hits.txt <span class="p">|</span> sort <span class="p">|</span> uniq &gt; summary_example/Kipp_etal_high_hit_subjects.txt
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># cut -f1 extracts the first field from the input file (e.g., HEADERS or individual read IDs), sorts all header names alphanumerically, and removes duplicates</span>
</span></span><span class="line"><span class="cl">	<span class="c1"># cut -f2 extracts the second field from the input file (e.g., SUBJECTS or where the read hit), sorts all subject names alphanumerical, and removes duplicates</span>
</span></span></code></pre></div><h3 id="separating-sequences-with-high-percentage-matching">Separating sequences with high percentage matching<a aria-label="Link to this section" class="paige-header-link" href="#separating-sequences-with-high-percentage-matching"></a></h3>
<p>Initial steps to list the headers of sequences needed:</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">zcat data_example/SRR17281099.fasta.gz &gt; data_example/SRR17281099.fasta
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#zcat unzips the zipped fasta file </span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">samtools faidx data_example/SRR17281099.fasta
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#samtools faidx indexes </span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nv">headers</span><span class="o">=</span><span class="sb">`</span>cat summary_example/Kipp_etal_high_hit_headers.txt<span class="sb">`</span>
</span></span></code></pre></div><p>Obtaining the subset of high perecentage matching sequences from the whole set of sequences.</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">samtools faidx data_example/SRR17281099.fasta <span class="nv">$headers</span> &gt; summary_example/Kipp_etal_high_hit_sequences.fasta
</span></span></code></pre></div><h3 id="step-7-explore-results-on-ncbi-blast-online">Step 7: Explore results on NCBI-BLAST (online)<a aria-label="Link to this section" class="paige-header-link" href="#step-7-explore-results-on-ncbi-blast-online"></a></h3>
<p>Below is a graph of the number of hits per reference pathogen sequence. Clearly, one sequence (<code>NC_021880.1_An_ph_str._JM</code>) is more common than the others.</p>
<img src="../results/hitsperwindows.jpg" alt="Alignment Process" style="max-width: 100%; height: auto; display: block; margin: 0 auto;"/>
<p>As an example, below a screenshot of the <code>Kipp_etal_blast_results_table.txt</code> output file. The data has been filtered based on % Match (e.g., percent identity) &gt;97% and sorted by descending read length. The longest, high-quality match was also from <code>NC_021990.1_An_ph_str._JM</code>. Many of the reads are 200bp or less, and while they may have a higher % Match, the longer reads tend to have higher E value scores, thus are more informative for comparisons to the entire NCBI database.</p>
<img src="../results/results2.png" alt="Alignment Process" style="max-width: 100%; height: auto; display: block; margin: 0 auto;"/>
<p>Next, copy the read name associated with the longest, highest-quality match (<code>SRR17281099.418555</code>). Open the text file <code>Kipp_etal_high_hit_sequences.fasta</code> and use <code>Ctrl-F</code> to find that read in the file.:</p>
<img src="../results/finder.jpg" alt="Alignment Process" style="max-width: 100%; height: auto; display: block; margin: 0 auto;"/>
<p>Once you&rsquo;ve located that read in the <code>Kipp_etal_high_hit_sequences.fasta</code> file, copy the sequence and paste the sequence into <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?PROGRAM=blastn&PAGE_TYPE=BlastSearch&LINK_LOC=blasthome">NCBI BLAST</a> online (Enter accession number(s), gi(s), or FASTA sequence(s) box in the upper left corner) and press <code>search</code>. This will return similar sequences in GenBank, displayed as shown in the image below:</p>
<p>First, we paste out sequence into the nucleotide box:
<img src="../results/blast1.jpg" alt="Alignment Process" style="max-width: 100%; height: auto; display: block; margin: 0 auto;"/></p>
<p>After searching, a results pane will show you results similar to what we had on the command line, but in a different format. In the top right panel, we can filter our hits by organism, percent identity, E value, and Query Coverage. The bottom panel shows all of the hits, sorted by E value by default. More information about each sequence that hit can be found by clicking the name in the Description column. There are also many options for viewing the alignments, such as in the Graphic Summary and Alignments tabs, or by clicking MSA viewer.</p>
<img src="../results/blast2.jpg" alt="Alignment Process" style="max-width: 100%; height: auto; display: block; margin: 0 auto;"/>
<p>NCBI has a neat feature where you can view how your sequence (highlighted in yellow) compares to other uploaded sequences:</p>
<img src="../results/blast3.jpg" alt="Alignment Process" style="max-width: 100%; height: auto; display: block; margin: 0 auto;"/>
<p>In this case, we can see that our longest read maps near-perfectly to a portion of an <a href="https://www.ncbi.nlm.nih.gov/nuccore/CP000235.1">Anaplasma phagocytophilum reference genome</a>, a Gram-negative bacterium that causes multiple diseases in cattle and sheep, as well as zoonoses like <a href="https://www.ncbi.nlm.nih.gov/books/NBK513341/">human granulocytic anaplasmosis.</a></p>
<br>
<hr>
<h1 id="real-time-example">Real-Time Example<a aria-label="Link to this section" class="paige-header-link" href="#real-time-example"></a></h1>
<p><strong>Description:</strong>
This section contains all of the code we will use to analyze the Nanopore sequence data generated during this workshop. This is not part of the worked example (above).
<br></p>
<p>What we choose to sequence will depend on the diversity of mammals collected, and will lead to one of two workflows: (1) Pathogen Detection via Host-Depletion or (2) Cryptic species identification via mitogenome enrichment</p>
<style>
  pre {
    font-size: 0.9em; /* Adjust to control the font size of the code */
    background-color: #2e2e2e; /* A subtle background for code */
    color: #f8f8f2;
    padding: 10px;
    border-radius: 4px;
  }

  h2, h3 {
    font-size: 1.5em;
  }

  .tab-button {
    padding: 10px 20px;
    background-color: #007BFF;
    color: black;
    border: none;
    cursor: pointer;
    margin-right: 10px;
    border-radius: 4px;
  }

  .tab-button:hover {
    background-color: #0056b3;
  }
</style>
<div style="margin-bottom: 20px;">
  <button id="windowsButton" class="tab-button" onclick="toggleSection('windows')">Pathogen Detection </button>
  <button id="macButton" class="tab-button" onclick="toggleSection('mac')">Cryptic Species Identification</button>
</div>
<!-- Windows (With Ubuntu) / Linux Installation Section -->
<div id="windows" style="display:none;">
<h2> Depletion of host genomes for bacterial and fungal pathogen identification  </h2>
<h2 id="1-basecalling">1. Basecalling<a aria-label="Link to this section" class="paige-header-link" href="#1-basecalling"></a></h2>
<p><strong>Description:</strong>
Basecalling is the process of determining the sequence of nucleotide bases from the raw data generated by the sequencer. For Nanopore sequencing, basecalling involves interpreting changes in electral current as single-stranded DNA (or RNA) passes through the nanopore. Current fluctuations correspond to different nucleotide bases which are decoded into <span style="color: #ff0000;">A&rsquo;s</span>,
<span style="color: #00ff00;">G&rsquo;s</span>,
<span style="color: #0000ff;">T&rsquo;s</span>,
and <span style="color: #ffa500;">C&rsquo;s</span> by the dorado basecalling software.</p>
<p>Bash is a scripting language that XYZ. We can bundle multiple commands together into a script (.sh) file, which is just a plain-text file that can be interpreted and executed by the computer. The code block below contains all of the text in the script file. All Bash scripts must begin with <code>#! /binbash/</code> as that line tells the computer where to find the Bash program/executable. Then, variables are defined (base_dir, output_dir, dorado) to make the actual code easier to read.</p>
<p>The code below runs the dorado basecaller on each barcdoe (1 through 24) and produces an output BAM file for each, respectively.</p>
<p><a href="../scripts/01_basecall.sh"><code>Download 01_basecall.sh</code></a></p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1"># Base directory for pod5 files</span>
</span></span><span class="line"><span class="cl"><span class="nv">base_dir</span><span class="o">=</span><span class="s2">&#34;/Library/MinKNOW/data/Ecuador_Workshop_Training_Pathogens_Pero/no_sample/20240913_1820_MN40667_FAY96875_60fc85db/pod5_pass&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Output directory for basecalling results</span>
</span></span><span class="line"><span class="cl"><span class="nv">output_dir</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Ecuador_Workshop/dorado-0.7.3-linux-x64/bin/01_sup_out/&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#dorado path</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nv">dorado</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Documents/software/dorado-0.7.0-osx-arm64/bin/dorado&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Loop through each barcode directory</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="k">for</span> barcode in barcode<span class="o">{</span>01..24<span class="o">}</span> 
</span></span><span class="line"><span class="cl"><span class="c1"># this loops over files named barcode01, barcode02, ..., barcode24</span>
</span></span><span class="line"><span class="cl"><span class="c1"># The number of barcode files will depend on the number of individuals multiplexed and being analyzed simultaneously</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="k">do</span>
</span></span><span class="line"><span class="cl">    <span class="nv">$dorado</span> basecaller dna_r10.4.1_e8.2_400bps_fast@v5.0.0 <span class="s2">&#34;</span><span class="nv">$base_dir</span><span class="s2">/</span><span class="nv">$barcode</span><span class="s2">/&#34;</span> &gt; <span class="s2">&#34;</span><span class="nv">$output_dir</span><span class="s2">/</span><span class="si">${</span><span class="nv">barcode</span><span class="si">}</span><span class="s2">.bam&#34;</span>
</span></span><span class="line"><span class="cl"><span class="k">done</span></span></span></code></pre></div>
<p>Names of Dorado models (i.e., Here, we use the basecaller model <a href="https://github.com/nanoporetech/dorado?tab=readme-ov-file#dna-models"><code>dna_r10.4.1_e8.2_400bps_fast@5.0.0</code></a>. Dorado model names are systematically structured, such that each segment corresponds to a different aspect of the model:</p>
<p><strong>Analyte Type (dna)</strong>: This denotes the type of analyte being sequenced. For DNA sequencing, it is represented as dna. If you are using a Direct RNA Sequencing Kit, this will be rna002 or rna004, depending on the kit.</p>
<p><strong>Pore Type (r10.4.1)</strong>: This section corresponds to the type of flow cell used. For instance, FLO-MIN114/FLO-FLG114 is indicated by r10.4.1, while FLO-MIN106D/FLO-FLG001 is signified by r9.4.1.</p>
<p><strong>Chemistry Type (e8.2)</strong>: This represents the chemistry type, which corresponds to the kit used for sequencing. For example, Kit 14 chemistry is denoted by e8.2 and Kit 10 or Kit 9 are denoted by e8.</p>
<p><strong>Translocation Speed (400bps)</strong>: This parameter, selected at the run setup in MinKNOW, refers to the speed of translocation. Prior to starting your run, a prompt will ask if you prefer to run at 260 bps or 400 bps. The former yields more accurate results but provides less data. As of MinKNOW version 23.04, the 260 bps option has been deprecated.</p>
<p><strong>Model Type (hac)</strong>: This represents the size of the model, where larger models yield more accurate basecalls but take more time. The three types of models are fast, hac, and sup. The fast model is the quickest, sup is the most accurate, and hac provides a balance between speed and accuracy. For most users, the hac model is recommended.</p>
<p><strong>Model Version Number (v5.0.0)</strong>: This denotes the version of the model. Model updates are regularly released, and higher version numbers typically signify greater accuracy.
For more details on decoding dorado model names see <a href="https://github.com/nanoporetech/dorado">here</a>.</p>
<br>
<h2 id="2-align-bam-files-to-a-reference-genome">2. Align BAM Files to a Reference Genome<a aria-label="Link to this section" class="paige-header-link" href="#2-align-bam-files-to-a-reference-genome"></a></h2>
<p>In this step, we align Nanopore reads to a reference genome. BAM files contain the same information as a SAM file, but in binary format which is not human readable. BAM files are generally smaller, however, and are therefore more efficient for software to work with. Alignment data is almost always stored as a BAM file and most software that analyzes aligned reads expects a BAM file as input.</p>
<p><a href="../scripts/02align.sh"><code>Download 02_align.sh</code></a></p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span>
</span></span><span class="line"><span class="cl"><span class="c1"># Path to the reference genome</span>
</span></span><span class="line"><span class="cl"><span class="nv">REFERENCE_PATH</span><span class="o">=</span><span class="s2">&#34;/Library/MinKNOW/data/References/Ecusdor_ref/pema_only_09092024.fasta&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Base directory for input BAM files </span>
</span></span><span class="line"><span class="cl"><span class="nv">INPUT_DIR</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Ecuador_Workshop/01_sup_out&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Output directory for aligned BAM files</span>
</span></span><span class="line"><span class="cl"><span class="nv">OUTPUT_DIR</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Ecuador_Workshop/02_align_out&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Loop over barcode01.bam to barcode24.bam</span>
</span></span><span class="line"><span class="cl"><span class="k">for</span> i in <span class="k">$(</span>seq -w <span class="m">1</span> 10<span class="k">)</span><span class="p">;</span> <span class="k">do</span>
</span></span><span class="line"><span class="cl">    <span class="c1"># Define input and output file names</span>
</span></span><span class="line"><span class="cl">    <span class="nv">INPUT_BAM</span><span class="o">=</span><span class="s2">&#34;</span><span class="si">${</span><span class="nv">INPUT_DIR</span><span class="si">}</span><span class="s2">/barcode</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span><span class="s2">.bam&#34;</span>
</span></span><span class="line"><span class="cl">    <span class="nv">OUTPUT_BAM</span><span class="o">=</span><span class="s2">&#34;</span><span class="si">${</span><span class="nv">OUTPUT_DIR</span><span class="si">}</span><span class="s2">/barcode</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span><span class="s2">_aligned.bam&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">    <span class="c1"># Run dorado aligner</span>
</span></span><span class="line"><span class="cl">    <span class="nb">echo</span> <span class="s2">&#34;Aligning </span><span class="si">${</span><span class="nv">INPUT_BAM</span><span class="si">}</span><span class="s2">...&#34;</span>
</span></span><span class="line"><span class="cl">    dorado aligner <span class="nv">$REFERENCE_PATH</span> <span class="nv">$INPUT_BAM</span> &gt; <span class="nv">$OUTPUT_BAM</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">    <span class="nb">echo</span> <span class="s2">&#34;Finished aligning barcode</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span><span class="s2">.bam&#34;</span>
</span></span><span class="line"><span class="cl"><span class="k">done</span></span></span></code></pre></div>
<br>
<h2 id="3-extracting-unaligned-reads">3. Extracting Unaligned Reads<a aria-label="Link to this section" class="paige-header-link" href="#3-extracting-unaligned-reads"></a></h2>
<p>Next, we extract the unaligned reads from the BAM files using <code>samtools</code>. This step is crucial for isolating reads that did not align to the reference genome, allowing us to focus on non-host sequences.</p>
<p><a href="../scripts/03_unmapped_reads.sh"><code>Download 03_unmapped_reads.sh</code></a></p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span>
</span></span><span class="line"><span class="cl"><span class="c1"># Path to samtools (ensure samtools is installed and accessible in your PATH)</span>
</span></span><span class="line"><span class="cl"><span class="nv">SAMTOOLS_PATH</span><span class="o">=</span><span class="s2">&#34;/opt/homebrew/bin/samtools&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Base directory for input BAM files (update this to the correct directory)</span>
</span></span><span class="line"><span class="cl"><span class="nv">INPUT_DIR</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Ecuador_Workshop/02_align_out&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Output directory for unaligned BAM files</span>
</span></span><span class="line"><span class="cl"><span class="nv">OUTPUT_DIR</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Ecuador_Workshop/03_unmapped_out&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Loop over barcode01.bam to barcode24.bam</span>
</span></span><span class="line"><span class="cl"><span class="k">for</span> i in <span class="k">$(</span>seq -w <span class="m">1</span> 24<span class="k">)</span><span class="p">;</span> <span class="k">do</span>
</span></span><span class="line"><span class="cl">    <span class="c1"># Define input and output file names</span>
</span></span><span class="line"><span class="cl">    <span class="nv">INPUT_BAM</span><span class="o">=</span><span class="s2">&#34;</span><span class="si">${</span><span class="nv">INPUT_DIR</span><span class="si">}</span><span class="s2">/barcode</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span><span class="s2">_aligned.bam&#34;</span>
</span></span><span class="line"><span class="cl">    <span class="nv">OUTPUT_BAM</span><span class="o">=</span><span class="s2">&#34;</span><span class="si">${</span><span class="nv">OUTPUT_DIR</span><span class="si">}</span><span class="s2">/barcode</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span><span class="s2">_unaligned.bam&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">    <span class="c1"># Extract unaligned reads using samtools</span>
</span></span><span class="line"><span class="cl">    <span class="nb">echo</span> <span class="s2">&#34;Extracting unaligned reads from </span><span class="si">${</span><span class="nv">INPUT_BAM</span><span class="si">}</span><span class="s2">...&#34;</span>
</span></span><span class="line"><span class="cl">    <span class="nv">$SAMTOOLS_PATH</span> view -b -f <span class="m">4</span> <span class="nv">$INPUT_BAM</span> &gt; <span class="nv">$OUTPUT_BAM</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">    <span class="nb">echo</span> <span class="s2">&#34;Finished extracting unaligned reads for barcode</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span><span class="s2">.bam&#34;</span>
</span></span><span class="line"><span class="cl"><span class="k">done</span></span></span></code></pre></div>
<br>
<h2 id="4-renaming-sequence-ids-before-database-creation">4. Renaming Sequence IDs Before Database Creation<a aria-label="Link to this section" class="paige-header-link" href="#4-renaming-sequence-ids-before-database-creation"></a></h2>
<p>Before creating the BLAST database, we rename the sequence IDs in the FASTA file to ensure uniqueness and compatibility. This step is essential for maintaining consistency in downstream analyses.</p>
<p><a href="../scripts/03.5_shorten_fasta_header.sh"><code>Download 03.5_shorten_fasta_header.sh</code></a></p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span>
</span></span><span class="line"><span class="cl">awk <span class="s1">&#39;
</span></span></span><span class="line"><span class="cl"><span class="s1">BEGIN {FS=&#34; &#34;; OFS=&#34; &#34;} 
</span></span></span><span class="line"><span class="cl"><span class="s1">/^&gt;/ {
</span></span></span><span class="line"><span class="cl"><span class="s1">    id=substr($1, 2, 40); # Get the first 40 characters of the ID
</span></span></span><span class="line"><span class="cl"><span class="s1">    count[id]++;          # Count occurrences of this ID
</span></span></span><span class="line"><span class="cl"><span class="s1">    if (count[id] &gt; 1) {  # If the ID is a duplicate, add a suffix
</span></span></span><span class="line"><span class="cl"><span class="s1">        id=id &#34;_&#34; count[id];
</span></span></span><span class="line"><span class="cl"><span class="s1">    }
</span></span></span><span class="line"><span class="cl"><span class="s1">    print &#34;&gt;&#34; id;         # Print the modified ID
</span></span></span><span class="line"><span class="cl"><span class="s1">    next;
</span></span></span><span class="line"><span class="cl"><span class="s1">}
</span></span></span><span class="line"><span class="cl"><span class="s1">{print}&#39;</span> /Users/mammalogy/Ecuador_Workshop/Pathogen_Ref/bvbrc_nsamer_good_comp_noplant_veupathbd_allbut_vecthost_pathogen_genomes_04092024.fasta &gt; /Users/mammalogy/Ecuador_Workshop/Pathogen_Ref/shortened_unique_ids.fasta</span></span></code></pre></div>
<br>
<h2 id="5-creating-a-blast-database">5. Creating a BLAST Database<a aria-label="Link to this section" class="paige-header-link" href="#5-creating-a-blast-database"></a></h2>
<p>With the sequence IDs renamed, we now create a BLAST database from the reference FASTA file. This database will be used for identifying the unaligned sequences.</p>
<p><a href="../scripts/04_blast_db_creation.sh"><code>Download 04_blast_db_creation.sh</code></a></p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span>
</span></span><span class="line"><span class="cl"><span class="c1"># Make the database</span>
</span></span><span class="line"><span class="cl">makeblastdb -in /Users/mammalogy/Ecuador_Workshop/Pathogen_Ref/shortened_unique_ids.fasta -dbtype nucl -out /Users/mammalogy/Ecuador_Workshop/Pathogen_Ref/bvbrc_pathogen_db -title <span class="s2">&#34;Pathogen Genomes DB&#34;</span> -parse_seqids
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Print success message</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s2">&#34;BLAST database created successfully: </span><span class="nv">$DB_NAME</span><span class="s2">&#34;</span></span></span></code></pre></div>
<br>
<h2 id="6-converting-bam-to-fasta">6. Converting BAM to FASTA<a aria-label="Link to this section" class="paige-header-link" href="#6-converting-bam-to-fasta"></a></h2>
<p>Before running BLAST, we need to convert the unaligned BAM files into FASTA format. This step prepares the sequences for the BLAST search.</p>
<p><a href="../scripts/05_convert_bam_to_fasta.sh"><code>Download 05_convert_bam_to_fasta.sh</code></a></p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span>
</span></span><span class="line"><span class="cl"><span class="c1"># Load samtools (if needed)</span>
</span></span><span class="line"><span class="cl"><span class="c1"># module load samtools</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Set paths</span>
</span></span><span class="line"><span class="cl"><span class="nv">BAM_DIR</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Ecuador_Workshop/03_unmapped_out&#34;</span>
</span></span><span class="line"><span class="cl"><span class="nv">FASTA_DIR</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Ecuador_Workshop/03_unmapped_fasta_out&#34;</span>
</span></span><span class="line"><span class="cl"><span class="nv">DB_NAME</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Ecuador_Workshop/Pathogen_Ref/bvbrc_pathogen_db&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Create output directory for FASTA files if it doesn&#39;t exist</span>
</span></span><span class="line"><span class="cl">mkdir -p <span class="nv">$FASTA_DIR</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Convert BAM files to FASTA</span>
</span></span><span class="line"><span class="cl"><span class="k">for</span> BAM_FILE in <span class="nv">$BAM_DIR</span>/*.bam<span class="p">;</span> <span class="k">do</span>
</span></span><span class="line"><span class="cl">    <span class="nv">BASENAME</span><span class="o">=</span><span class="k">$(</span>basename <span class="nv">$BAM_FILE</span> .bam<span class="k">)</span>
</span></span><span class="line"><span class="cl">    samtools fasta <span class="nv">$BAM_FILE</span> &gt; <span class="nv">$FASTA_DIR</span>/<span class="si">${</span><span class="nv">BASENAME</span><span class="si">}</span>.fasta
</span></span><span class="line"><span class="cl">    <span class="nb">echo</span> <span class="s2">&#34;Converted </span><span class="nv">$BAM_FILE</span><span class="s2"> to FASTA format&#34;</span>
</span></span><span class="line"><span class="cl"><span class="k">done</span></span></span></code></pre></div>
<br>
<h2 id="7-running-blast-on-fasta-files">7. Running BLAST on FASTA Files<a aria-label="Link to this section" class="paige-header-link" href="#7-running-blast-on-fasta-files"></a></h2>
<p>The converted FASTA files are then subjected to BLAST search against the created database to identify possible pathogenic sequences.</p>
<p><a href="../scripts/06_blast_fastas.sh"><code>Download 06_blast_fastas.sh</code></a></p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span>
</span></span><span class="line"><span class="cl"><span class="c1"># Set paths</span>
</span></span><span class="line"><span class="cl"><span class="nv">FASTA_DIR</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Ecuador_Workshop/03_unmapped_fasta_out&#34;</span>
</span></span><span class="line"><span class="cl"><span class="nv">DB_NAME</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Ecuador_Workshop/Pathogen_Ref/bvbrc_pathogen_db&#34;</span>
</span></span><span class="line"><span class="cl"><span class="nv">OUTPUT_DIR</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Ecuador_Workshop/03_blastn_results&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Create output directory for BLAST results if it doesn&#39;t exist</span>
</span></span><span class="line"><span class="cl">mkdir -p <span class="nv">$OUTPUT_DIR</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Run BLAST for each FASTA file</span>
</span></span><span class="line"><span class="cl"><span class="k">for</span> FASTA_FILE in <span class="nv">$FASTA_DIR</span>/*.fasta<span class="p">;</span> <span class="k">do</span>
</span></span><span class="line"><span class="cl">    <span class="nv">BASENAME</span><span class="o">=</span><span class="k">$(</span>basename <span class="nv">$FASTA_FILE</span> .fasta<span class="k">)</span>
</span></span><span class="line"><span class="cl">    blastn -query <span class="nv">$FASTA_FILE</span> -db <span class="nv">$DB_NAME</span> -out <span class="nv">$OUTPUT_DIR</span>/<span class="si">${</span><span class="nv">BASENAME</span><span class="si">}</span>_blastn.txt -outfmt <span class="m">6</span> -num_threads <span class="m">16</span>
</span></span><span class="line"><span class="cl">    <span class="nb">echo</span> <span class="s2">&#34;BLAST completed for </span><span class="nv">$FASTA_FILE</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl"><span class="k">done</span></span></span></code></pre></div>
<br>
<h2 id="8-renaming-sequence-ids-in-blast-results">8. Renaming Sequence IDs in BLAST Results<a aria-label="Link to this section" class="paige-header-link" href="#8-renaming-sequence-ids-in-blast-results"></a></h2>
<p>Finally, the sequence IDs in the BLAST output are renamed to match their original identifiers, providing clear traceability of the sequences.</p>
<p><a href="../scripts/07_rename_seqIDs.sh"><code>Download 07_rename_seqIDs.sh</code></a></p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span>
</span></span><span class="line"><span class="cl"><span class="c1"># Define directories and files</span>
</span></span><span class="line"><span class="cl"><span class="nv">RESULTS_DIR</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Ecuador_Workshop/03_blastn_results&#34;</span>
</span></span><span class="line"><span class="cl"><span class="nv">MAPPING_FILE</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Ecuador_Workshop/seq_id_mapping.txt&#34;</span>
</span></span><span class="line"><span class="cl"><span class="nv">OUTPUT_DIR</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Ecuador_Workshop/03_blastn_results_with_original_ids&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Create output directory if it doesn&#39;t exist</span>
</span></span><span class="line"><span class="cl">mkdir -p <span class="nv">$OUTPUT_DIR</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Loop through each BLAST result file from barcode01 to barcode24</span>
</span></span><span class="line"><span class="cl"><span class="k">for</span> i in <span class="k">$(</span>seq -f <span class="s2">&#34;%02g&#34;</span> <span class="m">1</span> 24<span class="k">)</span><span class="p">;</span> <span class="k">do</span>
</span></span><span class="line"><span class="cl">    <span class="nv">INPUT_FILE</span><span class="o">=</span><span class="s2">&#34;</span><span class="nv">$RESULTS_DIR</span><span class="s2">/barcode</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span><span class="s2">_unaligned_blastn.txt&#34;</span>
</span></span><span class="line"><span class="cl">    <span class="nv">OUTPUT_FILE</span><span class="o">=</span><span class="s2">&#34;</span><span class="nv">$OUTPUT_DIR</span><span class="s2">/barcode</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span><span class="s2">_unaligned_blastn_with_original_ids.txt&#34;</span>
</span></span><span class="line"><span class="cl">    
</span></span><span class="line"><span class="cl">    <span class="c1"># Replace seq_X IDs with original IDs using awk</span>
</span></span><span class="line"><span class="cl">    awk <span class="s1">&#39;NR==FNR {map[$1]=$2; next} $2 in map {$2=map[$2]}1&#39;</span> <span class="nv">$MAPPING_FILE</span> <span class="nv">$INPUT_FILE</span> &gt; <span class="nv">$OUTPUT_FILE</span>
</span></span><span class="line"><span class="cl">    
</span></span><span class="line"><span class="cl">    <span class="nb">echo</span> <span class="s2">&#34;Processed </span><span class="nv">$INPUT_FILE</span><span class="s2"> -&gt; </span><span class="nv">$OUTPUT_FILE</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl"><span class="k">done</span></span></span></code></pre></div>
</div>
<!--Cryptic Species ID -->
<div id="mac" style="display:none;">
  <h2>Molecular species ID through mitochondrial genome enrichment </h2>
<h2 id="1-basecalling-1">1. Basecalling<a aria-label="Link to this section" class="paige-header-link" href="#1-basecalling-1"></a></h2>
<p><strong>Description:</strong>
Basecalling is the process of determining the sequence of nucleotide bases from the raw data generated by the sequencer. For Nanopore sequencing, basecalling involves interpreting changes in electral current as single-stranded DNA (or RNA) passes through the nanopore. Current fluctuations correspond to different nucleotide bases which are decoded into <span style="color: #ff0000;">A&rsquo;s</span>,
<span style="color: #00ff00;">G&rsquo;s</span>,
<span style="color: #0000ff;">T&rsquo;s</span>,
and <span style="color: #ffa500;">C&rsquo;s</span> by the dorado basecalling software.</p>
<p>Bash is a scripting language that XYZ. We can bundle multiple commands together into a script (.sh) file, which is just a plain-text file that can be interpreted and executed by the computer. The code block below contains all of the text in the script file. All Bash scripts must begin with <code>#! /binbash/</code> as that line tells the computer where to find the Bash program/executable. Then, variables are defined (base_dir, output_dir, dorado) to make the actual code easier to read.</p>
<p>The code below runs the dorado basecaller on each barcdoe (1 through 24) and produces an output BAM file for each, respectively.</p>
<p><a href="../scripts/01_basecall.sh"><code>Download 01_basecall.sh</code></a></p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1"># Base directory for pod5 files</span>
</span></span><span class="line"><span class="cl"><span class="nv">base_dir</span><span class="o">=</span><span class="s2">&#34;/Library/MinKNOW/data/Ecuador_Workshop_Training_Pathogens_Pero/no_sample/20240913_1820_MN40667_FAY96875_60fc85db/pod5_pass&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Output directory for basecalling results</span>
</span></span><span class="line"><span class="cl"><span class="nv">output_dir</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/field_genomics/&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#dorado path</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nv">dorado</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Documents/software/dorado-0.7.0-osx-arm64/bin/dorado&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Loop through each barcode directory</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="k">for</span> barcode in barcode<span class="o">{</span>01..24<span class="o">}</span> 
</span></span><span class="line"><span class="cl"><span class="c1"># this loops over files named barcode01, barcode02, ..., barcode24</span>
</span></span><span class="line"><span class="cl"><span class="c1"># The number of barcode files will depend on the number of individuals multiplexed and being analyzed simultaneously</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="k">do</span>
</span></span><span class="line"><span class="cl">    <span class="nv">$dorado</span> basecaller dna_r10.4.1_e8.2_400bps_fast@v5.0.0 <span class="s2">&#34;</span><span class="nv">$base_dir</span><span class="s2">/</span><span class="nv">$barcode</span><span class="s2">/&#34;</span> &gt; <span class="s2">&#34;</span><span class="nv">$output_dir</span><span class="s2">/</span><span class="si">${</span><span class="nv">barcode</span><span class="si">}</span><span class="s2">.bam&#34;</span>
</span></span><span class="line"><span class="cl"><span class="k">done</span></span></span></code></pre></div>
<p>Names of Dorado models (i.e., Here, we use the basecaller model <a href="https://github.com/nanoporetech/dorado?tab=readme-ov-file#dna-models"><code>dna_r10.4.1_e8.2_400bps_fast@5.0.0</code></a>. Dorado model names are systematically structured, such that each segment corresponds to a different aspect of the model:</p>
<p><strong>Analyte Type (dna)</strong>: This denotes the type of analyte being sequenced. For DNA sequencing, it is represented as dna. If you are using a Direct RNA Sequencing Kit, this will be rna002 or rna004, depending on the kit.</p>
<p><strong>Pore Type (r10.4.1)</strong>: This section corresponds to the type of flow cell used. For instance, FLO-MIN114/FLO-FLG114 is indicated by r10.4.1, while FLO-MIN106D/FLO-FLG001 is signified by r9.4.1.</p>
<p><strong>Chemistry Type (e8.2)</strong>: This represents the chemistry type, which corresponds to the kit used for sequencing. For example, Kit 14 chemistry is denoted by e8.2 and Kit 10 or Kit 9 are denoted by e8.</p>
<p><strong>Translocation Speed (400bps)</strong>: This parameter, selected at the run setup in MinKNOW, refers to the speed of translocation. Prior to starting your run, a prompt will ask if you prefer to run at 260 bps or 400 bps. The former yields more accurate results but provides less data. As of MinKNOW version 23.04, the 260 bps option has been deprecated.</p>
<p><strong>Model Type (hac)</strong>: This represents the size of the model, where larger models yield more accurate basecalls but take more time. The three types of models are fast, hac, and sup. The fast model is the quickest, sup is the most accurate, and hac provides a balance between speed and accuracy. For most users, the hac model is recommended.</p>
<p><strong>Model Version Number (v5.0.0)</strong>: This denotes the version of the model. Model updates are regularly released, and higher version numbers typically signify greater accuracy.
For more details on decoding dorado model names see <a href="https://github.com/nanoporetech/dorado">here</a>.</p>
<h2 id="2-convert-each-bam-file-to-a-fasta-format">2. Convert each BAM file to a FASTA format<a aria-label="Link to this section" class="paige-header-link" href="#2-convert-each-bam-file-to-a-fasta-format"></a></h2>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1"># Base directory for output</span>
</span></span><span class="line"><span class="cl"><span class="nv">base_dir</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Ecuador_Workshop/01_sup_out&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Output directory for basecalling results</span>
</span></span><span class="line"><span class="cl"><span class="nv">output_dir</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Ecuador_Workshop/02_convert_bams_out/&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">mkdir <span class="nv">$output_dir</span> -p
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="k">for</span> barcode in barcode<span class="o">{</span>01..24<span class="o">}</span> 
</span></span><span class="line"><span class="cl"><span class="c1"># this loops over files named barcode01, barcode02, ..., barcode24</span>
</span></span><span class="line"><span class="cl"><span class="c1"># The number of barcode files will depend on the number of individuals multiplexed and being analyzed simultaneously</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="k">do</span>
</span></span><span class="line"><span class="cl">    samtools fastq <span class="s2">&#34;</span><span class="nv">$base_dir</span><span class="s2">/</span><span class="nv">$barcode</span><span class="s2">.bam&#34;</span> &gt; <span class="s2">&#34;</span><span class="nv">$output_dir</span><span class="s2">/</span><span class="si">${</span><span class="nv">barcode</span><span class="si">}</span><span class="s2">.fastq&#34;</span>
</span></span><span class="line"><span class="cl"><span class="k">done</span></span></span></code></pre></div>
<br>
</div>
<h2 id="2-run-flye">2. Run Flye<a aria-label="Link to this section" class="paige-header-link" href="#2-run-flye"></a></h2>
<p><strong>Description</strong>:
<a href="https://github.com/mikolmogorov/Flye">FLYE</a> is a software tool used to assemble genomes from long DNA sequences without a reference file to base the alignments off of. It pieces together overlapping DNA reads to form a continuous sequence, even if the data is noisy or has gaps. For mitogenomes, FLYE helps combine these long reads into a complete mitochondrial genome, and sometimes reads may span the entire mitogenome, making FLYE&rsquo;s aligning pretty easy.</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1"># Base directory for output</span>
</span></span><span class="line"><span class="cl"><span class="nv">base_dir</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Ecuador_Workshop/02_convert_bams_out&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Output directory for basecalling results</span>
</span></span><span class="line"><span class="cl"><span class="nv">output_dir</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Ecuador_Workshop/03_flye_out&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">mkdir <span class="nv">$output_dir</span> -p
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">conda activate base
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="k">for</span> barcode in barcode<span class="o">{</span>01..24<span class="o">}</span> 
</span></span><span class="line"><span class="cl"><span class="c1"># this loops over files named barcode01, barcode02, ..., barcode24</span>
</span></span><span class="line"><span class="cl"><span class="c1"># The number of barcode files will depend on the number of individuals multiplexed and being analyzed simultaneously</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="k">do</span>
</span></span><span class="line"><span class="cl">  flye --genome-size <span class="m">16000</span> --nano-hq <span class="nv">$base_dir</span>/<span class="nv">$barcode</span>.fastq --out-dir <span class="nv">$output_dir</span>/<span class="nv">$barcode</span>
</span></span><span class="line"><span class="cl"><span class="k">done</span></span></span></code></pre></div>
<br>
</div>
<h2 id="3-align-your-mitochondrial-genomes-using-dorado">3. Align your mitochondrial genomes using Dorado<a aria-label="Link to this section" class="paige-header-link" href="#3-align-your-mitochondrial-genomes-using-dorado"></a></h2>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1"># Base directory for output</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nv">dorado</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Documents/software/dorado-0.7.0-osx-arm64/bin/dorado&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nv">base_dir</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Ecuador_Workshop/01_sup_out&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Output directory for basecalling results</span>
</span></span><span class="line"><span class="cl"><span class="nv">output_dir</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Ecuador_Workshop/03_dorado_align_out/&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nv">reference_path</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Ecuador_Workshop/Peromyscus_mito/pero_mito.txt&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">mkdir <span class="nv">$output_dir</span> -p
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="k">for</span> barcode in barcode<span class="o">{</span>01..24<span class="o">}</span> 
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="k">do</span>
</span></span><span class="line"><span class="cl">  <span class="nv">$dorado</span> aligner <span class="nv">$reference_path</span> <span class="nv">$base_dir</span>/<span class="nv">$barcode</span>.bam <span class="p">|</span> samtools sort &gt; <span class="nv">$output_dir</span>/<span class="nv">$barcode</span>.bam 
</span></span><span class="line"><span class="cl"><span class="k">done</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># The &#34; | &#34; is a pipe command. It sends the output of the alignment into the samtools sort funtion. This function puts our reads in order so we can create a consensus sequence on the next step. </span></span></span></code></pre></div>
<h2 id="create-a-consensus-sequence-using-samtools">Create a Consensus Sequence using samtools<a aria-label="Link to this section" class="paige-header-link" href="#create-a-consensus-sequence-using-samtools"></a></h2>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="k">for</span> barcode in barcode<span class="o">{</span>01..24<span class="o">}</span> 
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="k">do</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Pull consensus sequence sequence of the sample out of the alignment (BAM) file</span>
</span></span><span class="line"><span class="cl">  samtools consensus -a --min-MQ <span class="m">10</span> --min-BQ <span class="m">10</span> <span class="nv">$output_dir</span>/<span class="nv">$barcode</span>.bam &gt; <span class="nv">$output_dir</span>/<span class="nv">$barcode</span>.fasta
</span></span><span class="line"><span class="cl"><span class="c1">#-a include all positions</span>
</span></span><span class="line"><span class="cl"><span class="c1">#-min-MQ minimum mapping quality</span>
</span></span><span class="line"><span class="cl"><span class="c1">#-min-BQ minimum base quality</span>
</span></span><span class="line"><span class="cl"><span class="k">done</span></span></span></code></pre></div>
<h2 id="converting-our-consensus-sequence-into-a-format-for-alingment">Converting our consensus sequence into a format for alingment<a aria-label="Link to this section" class="paige-header-link" href="#converting-our-consensus-sequence-into-a-format-for-alingment"></a></h2>
<h1 id="rename-the-header-in-each-fasta-file-by-the-name-of-the-fasta-file">Rename the header in each fasta file by the name of the fasta file<a aria-label="Link to this section" class="paige-header-link" href="#rename-the-header-in-each-fasta-file-by-the-name-of-the-fasta-file"></a></h1>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="k">for</span> barcode in barcode<span class="o">{</span>01..24<span class="o">}</span> 
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="k">do</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">sed <span class="s1">&#39;s/^&gt;.*/&gt;&#39;</span><span class="si">${</span><span class="nv">barcode</span><span class="si">}</span><span class="s1">&#39;/&#39;</span> <span class="nv">$output_dir</span>/<span class="nv">$barcode</span>.fasta &gt; <span class="nv">$output_dir</span>/<span class="nv">$barcode</span>.rename.fasta
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="k">done</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># concatanate all mitogenome sequences into one fasta file</span>
</span></span><span class="line"><span class="cl">cat <span class="nv">$output_dir</span>/*.rename.fasta &gt; <span class="nv">$output_dir</span>/concatanated.fasta
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Use MEGA to bring in concatanated fasta file, align the sequences, and build a tree</span></span></span></code></pre></div>
</main>
        

        









    

    




    </div>
</article>






























    
    



    



    



    



    



    



    



    



    



    






                    











    





    <footer id="paige-site-footer">
        

        
            <p class="paige-row-narrow text-center text-secondary" id="paige-copyright">Â© Alex Hey</p>
        

        

        

        
    </footer>



                    
                </div>
            </div>
        </div>

        















    
    
    


<script  crossorigin="anonymous"   defer  integrity="sha256-gfPiYYSP&#43;RNtNeRBt45mJombFJmNf0V0Ipu785Obi0M="   referrerpolicy="no-referrer"  src="../js/paige/bootstrap/bootstrap.bundle.min.81f3e261848ff9136d35e441b78e6626899b14998d7f4574229bbbf3939b8b43.js" ></script>







    



    



    



    



    



    



    



    



    



    



    



    



    



    



    





<noscript>JavaScript is required</noscript>


        
    </body>
</html>
