

<!doctype html>






<html  data-paige="Paige theme from https://github.com/willfaught/paige"   lang="en-us" >

    










    

    

    




<head>
    

    

<meta charset="utf-8">







<meta content="#0d6efd" name="msapplication-TileColor">


    <meta content="/browserconfig.xml" name="msapplication-config">


<meta content="https://github.com/willfaught/paige" name="theme">
<meta content="#0d6efd" name="theme-color">
<meta content="width=device-width, initial-scale=1" name="viewport">

<meta property="og:url" content="/workflows/">
  <meta property="og:title" content="Nanopore workflow">
  <meta property="og:description" content=" Table of Contents Molecular Procedures Worked Example Real-Time Example Molecular Procedures DNA Extraction Library Preparation Flow Cell Wash and Storage DNA Extraction Download DNA Extraction Protocol Library Preparation Download Nanopore Native Barcode Library Prep PDF Flow Cell Wash and Storage Download Flow Cell Wash and Storage Protocol PDF Worked Example The goal of this portion of the workshop is to practice the basic bioinformatic workflow required to process nanopore adaptive sampling sequence data. Specifically, we will map reads to reference sequences, filter out host sequences, and perform homology searches against known pathogen sequences. This workflow uses publicly available data published by Kipp et al. (2023).">
  <meta property="og:locale" content="en_us">
  <meta property="og:type" content="website">


  <meta name="twitter:card" content="summary">
  <meta name="twitter:title" content="Nanopore workflow">
  <meta name="twitter:description" content=" Table of Contents Molecular Procedures Worked Example Real-Time Example Molecular Procedures DNA Extraction Library Preparation Flow Cell Wash and Storage DNA Extraction Download DNA Extraction Protocol Library Preparation Download Nanopore Native Barcode Library Prep PDF Flow Cell Wash and Storage Download Flow Cell Wash and Storage Protocol PDF Worked Example The goal of this portion of the workshop is to practice the basic bioinformatic workflow required to process nanopore adaptive sampling sequence data. Specifically, we will map reads to reference sequences, filter out host sequences, and perform homology searches against known pathogen sequences. This workflow uses publicly available data published by Kipp et al. (2023).">



    
        <title>Nanopore workflow Â· Introduction</title>
    

    


    <link href="../apple-touch-icon.png" rel="apple-touch-icon" sizes="180x180">



    <link href="../favicon.ico" rel="shortcut icon">



    <link href="../favicon.png" rel="icon" type="image/png">



    <link href="../favicon.svg" rel="icon" type="image/svg+xml">



    <link href="../favicon-16x16.png" rel="icon" sizes="16x16" type="image/png">



    <link href="../favicon-32x32.png" rel="icon" sizes="32x32" type="image/png">



    <link color="#0d6efd" href="../safari-pinned-tab.svg" rel="mask-icon">



    <link href="../site.webmanifest" rel="manifest">
























    

    
        
        
        
    

    
        
    

    
    
    


<link  crossorigin="anonymous"  href="../css/paige/bootstrap/6186ad141d391e4c0c10d2e9cb4eaeaadf75e809-paige.min.90e8e43521c2ef40c030c3979ea72bfc4a04397b9995dce5eadbe91e9770a96b.css"  integrity="sha256-kOjkNSHC70DAMMOXnqcr/EoEOXuZldzl6tvpHpdwqWs="   referrerpolicy="no-referrer"  rel="stylesheet">



















    

    

    

    
    
    


<link  crossorigin="anonymous"  href="../css/paige/bootstrap-icons/bootstrap-icons.min.540116b7267d4b2410c74086ae6497566928c6d5715900be7ce7279ad45addcd.css"  integrity="sha256-VAEWtyZ9SyQQx0CGrmSXVmkoxtVxWQC&#43;fOcnmtRa3c0="   referrerpolicy="no-referrer"  rel="stylesheet">






    

<style>

    


.paige-figure-numbered {
    counter-increment: paige-figure-numbered;
}

.paige-figure-numbered > div > figure > figcaption::before {
    content: "Figure " counter(paige-figure-numbered) ": ";
}

.paige-figure-numbered > div > figure > figcaption:empty::before {
    content: "Figure " counter(paige-figure-numbered);
}

.paige-header-link {
    opacity: 0;
    margin-left: 0.5rem;
    position: absolute;
    transition: color 0.15s ease-in-out, opacity 0.15s ease-in-out;
}

.paige-header-link::after {
    content: "#";
}

.paige-quote .blockquote-footer {
    margin-top: 0;
}

.paige-quote blockquote {
    border-left: 0;
    border-right: 0;
    margin-bottom: 0;
    padding: 0;
}

blockquote {
    padding: 0.5rem 1rem;
    border-left: 0.25rem solid var(--bs-border-color);
    border-right: 0.25rem solid var(--bs-body-bg);
}

td,
th {
    padding: 0.25rem;
}

.highlight .chroma .hl,
.highlight .chroma .lnt {
    display: flex;
}

.paige-row-narrow:not(:last-child),
.paige-figure .paige-quote,
.paige-figure .paige-video,
.paige-figure .highlight pre.chroma,
.paige-figure .highlight .chroma pre,
.paige-figure .paige-quote blockquote,
.paige-figure figure > div > :last-child,
.paige-gallery .paige-figure,
.paige-gallery .paige-image,
blockquote > p:last-of-type {
    margin-bottom: 0;
}

.paige-figure,
.paige-gallery,
.paige-image,
.paige-quote,
.paige-row-narrow:last-child,
.paige-row-wide,
.paige-video,
#paige-content > *,
table {
    margin-bottom: 1rem;
}

.paige-header-link:focus,
.paige-header-link:hover,
:hover > .paige-header-link,
:target > .paige-header-link {
    opacity: 1;
}




    @media (prefers-color-scheme: dark) {
        /* Background */ .bg { color: #c9d1d9; background-color: var(--bs-body-bg); }
/* PreWrapper */ .chroma { color: #c9d1d9; background-color: var(--bs-body-bg); }
/* Other */ .chroma .x {  }
/* Error */ .chroma .err { color: #f85149 }
/* CodeLine */ .chroma .cl {  }
/* LineLink */ .chroma .lnlinks { outline: none; text-decoration: none; color: inherit }
/* LineTableTD */ .chroma .lntd { vertical-align: top; padding: 0; margin: 0; border: 0; }
/* LineTable */ .chroma .lntable { border-spacing: 0; padding: 0; margin: 0; border: 0; }
/* LineHighlight */ .chroma .hl { background-color: #4f4f4d }
/* LineNumbersTable */ .chroma .lnt { white-space: pre; user-select: none; margin-right: 0.4em; padding: 0 0.4em 0 0.4em;color: #64686c }
/* LineNumbers */ .chroma .ln { white-space: pre; user-select: none; margin-right: 0.4em; padding: 0 0.4em 0 0.4em;color: #6e7681 }
/* Line */ .chroma .line { display: flex; }
/* Keyword */ .chroma .k { color: #ff7b72 }
/* KeywordConstant */ .chroma .kc { color: #79c0ff }
/* KeywordDeclaration */ .chroma .kd { color: #ff7b72 }
/* KeywordNamespace */ .chroma .kn { color: #ff7b72 }
/* KeywordPseudo */ .chroma .kp { color: #79c0ff }
/* KeywordReserved */ .chroma .kr { color: #ff7b72 }
/* KeywordType */ .chroma .kt { color: #ff7b72 }
/* Name */ .chroma .n {  }
/* NameAttribute */ .chroma .na {  }
/* NameBuiltin */ .chroma .nb {  }
/* NameBuiltinPseudo */ .chroma .bp {  }
/* NameClass */ .chroma .nc { color: #f0883e; font-weight: bold }
/* NameConstant */ .chroma .no { color: #79c0ff; font-weight: bold }
/* NameDecorator */ .chroma .nd { color: #d2a8ff; font-weight: bold }
/* NameEntity */ .chroma .ni { color: #ffa657 }
/* NameException */ .chroma .ne { color: #f0883e; font-weight: bold }
/* NameFunction */ .chroma .nf { color: #d2a8ff; font-weight: bold }
/* NameFunctionMagic */ .chroma .fm {  }
/* NameLabel */ .chroma .nl { color: #79c0ff; font-weight: bold }
/* NameNamespace */ .chroma .nn { color: #ff7b72 }
/* NameOther */ .chroma .nx {  }
/* NameProperty */ .chroma .py { color: #79c0ff }
/* NameTag */ .chroma .nt { color: #7ee787 }
/* NameVariable */ .chroma .nv { color: #79c0ff }
/* NameVariableClass */ .chroma .vc {  }
/* NameVariableGlobal */ .chroma .vg {  }
/* NameVariableInstance */ .chroma .vi {  }
/* NameVariableMagic */ .chroma .vm {  }
/* Literal */ .chroma .l { color: #a5d6ff }
/* LiteralDate */ .chroma .ld { color: #79c0ff }
/* LiteralString */ .chroma .s { color: #a5d6ff }
/* LiteralStringAffix */ .chroma .sa { color: #79c0ff }
/* LiteralStringBacktick */ .chroma .sb { color: #a5d6ff }
/* LiteralStringChar */ .chroma .sc { color: #a5d6ff }
/* LiteralStringDelimiter */ .chroma .dl { color: #79c0ff }
/* LiteralStringDoc */ .chroma .sd { color: #a5d6ff }
/* LiteralStringDouble */ .chroma .s2 { color: #a5d6ff }
/* LiteralStringEscape */ .chroma .se { color: #79c0ff }
/* LiteralStringHeredoc */ .chroma .sh { color: #79c0ff }
/* LiteralStringInterpol */ .chroma .si { color: #a5d6ff }
/* LiteralStringOther */ .chroma .sx { color: #a5d6ff }
/* LiteralStringRegex */ .chroma .sr { color: #79c0ff }
/* LiteralStringSingle */ .chroma .s1 { color: #a5d6ff }
/* LiteralStringSymbol */ .chroma .ss { color: #a5d6ff }
/* LiteralNumber */ .chroma .m { color: #a5d6ff }
/* LiteralNumberBin */ .chroma .mb { color: #a5d6ff }
/* LiteralNumberFloat */ .chroma .mf { color: #a5d6ff }
/* LiteralNumberHex */ .chroma .mh { color: #a5d6ff }
/* LiteralNumberInteger */ .chroma .mi { color: #a5d6ff }
/* LiteralNumberIntegerLong */ .chroma .il { color: #a5d6ff }
/* LiteralNumberOct */ .chroma .mo { color: #a5d6ff }
/* Operator */ .chroma .o { color: #ff7b72; font-weight: bold }
/* OperatorWord */ .chroma .ow { color: #ff7b72; font-weight: bold }
/* Punctuation */ .chroma .p {  }
/* Comment */ .chroma .c { color: #8b949e; font-style: italic }
/* CommentHashbang */ .chroma .ch { color: #8b949e; font-style: italic }
/* CommentMultiline */ .chroma .cm { color: #8b949e; font-style: italic }
/* CommentSingle */ .chroma .c1 { color: #8b949e; font-style: italic }
/* CommentSpecial */ .chroma .cs { color: #8b949e; font-weight: bold; font-style: italic }
/* CommentPreproc */ .chroma .cp { color: #8b949e; font-weight: bold; font-style: italic }
/* CommentPreprocFile */ .chroma .cpf { color: #8b949e; font-weight: bold; font-style: italic }
/* Generic */ .chroma .g {  }
/* GenericDeleted */ .chroma .gd { color: #ffa198; background-color: #490202 }
/* GenericEmph */ .chroma .ge { font-style: italic }
/* GenericError */ .chroma .gr { color: #ffa198 }
/* GenericHeading */ .chroma .gh { color: #79c0ff; font-weight: bold }
/* GenericInserted */ .chroma .gi { color: #56d364; background-color: #0f5323 }
/* GenericOutput */ .chroma .go { color: #8b949e }
/* GenericPrompt */ .chroma .gp { color: #8b949e }
/* GenericStrong */ .chroma .gs { font-weight: bold }
/* GenericSubheading */ .chroma .gu { color: #79c0ff }
/* GenericTraceback */ .chroma .gt { color: #ff7b72 }
/* GenericUnderline */ .chroma .gl { text-decoration: underline }
/* TextWhitespace */ .chroma .w { color: #6e7681 }

    }

    @media (prefers-color-scheme: light) {
        /* Background */ .bg { background-color: #ffffff; }
/* PreWrapper */ .chroma { background-color: #ffffff; }
/* Other */ .chroma .x {  }
/* Error */ .chroma .err { color: #a61717; background-color: #e3d2d2 }
/* CodeLine */ .chroma .cl {  }
/* LineLink */ .chroma .lnlinks { outline: none; text-decoration: none; color: inherit }
/* LineTableTD */ .chroma .lntd { vertical-align: top; padding: 0; margin: 0; border: 0; }
/* LineTable */ .chroma .lntable { border-spacing: 0; padding: 0; margin: 0; border: 0; }
/* LineHighlight */ .chroma .hl { background-color: #ffffcc }
/* LineNumbersTable */ .chroma .lnt { white-space: pre; user-select: none; margin-right: 0.4em; padding: 0 0.4em 0 0.4em;color: #7f7f7f }
/* LineNumbers */ .chroma .ln { white-space: pre; user-select: none; margin-right: 0.4em; padding: 0 0.4em 0 0.4em;color: #7f7f7f }
/* Line */ .chroma .line { display: flex; }
/* Keyword */ .chroma .k { color: #000000; font-weight: bold }
/* KeywordConstant */ .chroma .kc { color: #000000; font-weight: bold }
/* KeywordDeclaration */ .chroma .kd { color: #000000; font-weight: bold }
/* KeywordNamespace */ .chroma .kn { color: #000000; font-weight: bold }
/* KeywordPseudo */ .chroma .kp { color: #000000; font-weight: bold }
/* KeywordReserved */ .chroma .kr { color: #000000; font-weight: bold }
/* KeywordType */ .chroma .kt { color: #445588; font-weight: bold }
/* Name */ .chroma .n {  }
/* NameAttribute */ .chroma .na { color: #008080 }
/* NameBuiltin */ .chroma .nb { color: #0086b3 }
/* NameBuiltinPseudo */ .chroma .bp { color: #999999 }
/* NameClass */ .chroma .nc { color: #445588; font-weight: bold }
/* NameConstant */ .chroma .no { color: #008080 }
/* NameDecorator */ .chroma .nd { color: #3c5d5d; font-weight: bold }
/* NameEntity */ .chroma .ni { color: #800080 }
/* NameException */ .chroma .ne { color: #990000; font-weight: bold }
/* NameFunction */ .chroma .nf { color: #990000; font-weight: bold }
/* NameFunctionMagic */ .chroma .fm {  }
/* NameLabel */ .chroma .nl { color: #990000; font-weight: bold }
/* NameNamespace */ .chroma .nn { color: #555555 }
/* NameOther */ .chroma .nx {  }
/* NameProperty */ .chroma .py {  }
/* NameTag */ .chroma .nt { color: #000080 }
/* NameVariable */ .chroma .nv { color: #008080 }
/* NameVariableClass */ .chroma .vc { color: #008080 }
/* NameVariableGlobal */ .chroma .vg { color: #008080 }
/* NameVariableInstance */ .chroma .vi { color: #008080 }
/* NameVariableMagic */ .chroma .vm {  }
/* Literal */ .chroma .l {  }
/* LiteralDate */ .chroma .ld {  }
/* LiteralString */ .chroma .s { color: #dd1144 }
/* LiteralStringAffix */ .chroma .sa { color: #dd1144 }
/* LiteralStringBacktick */ .chroma .sb { color: #dd1144 }
/* LiteralStringChar */ .chroma .sc { color: #dd1144 }
/* LiteralStringDelimiter */ .chroma .dl { color: #dd1144 }
/* LiteralStringDoc */ .chroma .sd { color: #dd1144 }
/* LiteralStringDouble */ .chroma .s2 { color: #dd1144 }
/* LiteralStringEscape */ .chroma .se { color: #dd1144 }
/* LiteralStringHeredoc */ .chroma .sh { color: #dd1144 }
/* LiteralStringInterpol */ .chroma .si { color: #dd1144 }
/* LiteralStringOther */ .chroma .sx { color: #dd1144 }
/* LiteralStringRegex */ .chroma .sr { color: #009926 }
/* LiteralStringSingle */ .chroma .s1 { color: #dd1144 }
/* LiteralStringSymbol */ .chroma .ss { color: #990073 }
/* LiteralNumber */ .chroma .m { color: #009999 }
/* LiteralNumberBin */ .chroma .mb { color: #009999 }
/* LiteralNumberFloat */ .chroma .mf { color: #009999 }
/* LiteralNumberHex */ .chroma .mh { color: #009999 }
/* LiteralNumberInteger */ .chroma .mi { color: #009999 }
/* LiteralNumberIntegerLong */ .chroma .il { color: #009999 }
/* LiteralNumberOct */ .chroma .mo { color: #009999 }
/* Operator */ .chroma .o { color: #000000; font-weight: bold }
/* OperatorWord */ .chroma .ow { color: #000000; font-weight: bold }
/* Punctuation */ .chroma .p {  }
/* Comment */ .chroma .c { color: #999988; font-style: italic }
/* CommentHashbang */ .chroma .ch { color: #999988; font-style: italic }
/* CommentMultiline */ .chroma .cm { color: #999988; font-style: italic }
/* CommentSingle */ .chroma .c1 { color: #999988; font-style: italic }
/* CommentSpecial */ .chroma .cs { color: #999999; font-weight: bold; font-style: italic }
/* CommentPreproc */ .chroma .cp { color: #999999; font-weight: bold; font-style: italic }
/* CommentPreprocFile */ .chroma .cpf { color: #999999; font-weight: bold; font-style: italic }
/* Generic */ .chroma .g {  }
/* GenericDeleted */ .chroma .gd { color: #000000; background-color: #ffdddd }
/* GenericEmph */ .chroma .ge { color: #000000; font-style: italic }
/* GenericError */ .chroma .gr { color: #aa0000 }
/* GenericHeading */ .chroma .gh { color: #999999 }
/* GenericInserted */ .chroma .gi { color: #000000; background-color: #ddffdd }
/* GenericOutput */ .chroma .go { color: #888888 }
/* GenericPrompt */ .chroma .gp { color: #555555 }
/* GenericStrong */ .chroma .gs { font-weight: bold }
/* GenericSubheading */ .chroma .gu { color: #aaaaaa }
/* GenericTraceback */ .chroma .gt { color: #aa0000 }
/* GenericUnderline */ .chroma .gl { text-decoration: underline }
/* TextWhitespace */ .chroma .w { color: #bbbbbb }

    }


@media (prefers-reduced-motion: reduce) {
    .paige-header-link {
        transition: none;
    }
}




</style>


    
</head>

    <body>

        

        <div class="container">
            <div class="row">
                <div class="col mt-3" id="paige-site">
                    

                    














    





    



    <header id="paige-site-header">
        

        
            <div class="display-1 fw-bold  mb-2  paige-row-wide text-center" id="paige-site-title">Field Genomics Workshop</div>
        

        
            <div class="lead paige-row-wide text-center" id="paige-site-description">Ecuador 2024</div>
        

        
            
                <nav aria-label="Menu" class="paige-row-wide" id="paige-menu">
                    <ul class="align-items-center justify-content-center nav ">
                        
                            
                            
                            
                            
                            

                            <li class="nav-item ">
                                <a   class="  nav-link   text-decoration-underline "  href="../"    >Home</a>

                                
                            </li>
                        
                            
                            
                            
                            
                            

                            <li class="nav-item ">
                                <a   class="  nav-link   text-decoration-underline "  href="../getting-ready/"    >Getting Ready</a>

                                
                            </li>
                        
                            
                            
                            
                            
                            

                            <li class="nav-item ">
                                <a  aria-current="page"   class=" active  fw-bold text-body-emphasis    nav-link   text-decoration-underline "  href="../workflows/"    >Workflows</a>

                                
                            </li>
                        
                            
                            
                            
                            
                            

                            <li class="nav-item ">
                                <a   class="  nav-link   text-decoration-underline "  href="../slides/"    >Slides</a>

                                
                            </li>
                        
                            
                            
                            
                            
                            

                            <li class="nav-item ">
                                <a   class="  nav-link   text-decoration-underline "  href="../replicate-this-workshop/"    >Replicate This Workshop</a>

                                
                            </li>
                        
                            
                            
                            
                            
                            

                            <li class="nav-item ">
                                <a   class="  nav-link   text-decoration-underline "  href="../contactus/"    >Contact Us</a>

                                
                            </li>
                        
                    </ul>
                </nav>
            
        

        

        
    </header>


                    




















    



    
        
    




<article class="paige-kind-section paige-published" id="paige-page">
    <div class="align-items-center d-flex flex-column">
        




















    <header class="mw-100" id="paige-page-header">
        <div class="align-items-center d-flex flex-column">
            

            
                <h1 class="fw-bold" id="paige-page-title">Nanopore workflow</h1>
            

            

            

            

            

            
        </div>
    </header>



        
            <main class="mw-100" id="paige-content"></div>
<hr>
<h2 id="table-of-contents">Table of Contents<a aria-label="Link to this section" class="paige-header-link" href="#table-of-contents"></a></h2>
<ul>
<li><a href="#molecular-procedures">Molecular Procedures</a></li>
<li><a href="#worked-example">Worked Example</a></li>
<li><a href="#real-time-example">Real-Time Example</a></li>
</ul>
<br>
<h2 id="molecular-procedures">Molecular Procedures<a aria-label="Link to this section" class="paige-header-link" href="#molecular-procedures"></a></h2>
<!-- Buttons to toggle sections -->
<div style="text-align: center; margin-bottom: 20px;">
  <button onclick="toggleSection('dnaExtraction')"> DNA Extraction</button>
  <button onclick="toggleSection('libraryPrep')">Library Preparation</button>
  <button onclick="toggleSection('flowCell')"> Flow Cell Wash and Storage</button>
</div>
<!-- Containers for the PDFs -->
<div style="display: flex; flex-wrap: wrap; gap: 20px; justify-content: space-evenly;">
  <!-- DNA Extraction -->
  <div id="dnaExtraction" class="pdf-section" style="display: none; border: 1px solid #ccc; border-radius: 8px; width: 600px; padding: 15px; box-shadow: 2px 2px 8px rgba(0,0,0,0.1); text-align: center;">
    <h3>DNA Extraction</h3>
    <a href="../procedures/extraction.pdf" target="_blank">Download DNA Extraction Protocol</a>
    <iframe src="../procedures/extraction.pdf" width="100%" height="400" style="border: none; margin-top: 10px;"></iframe>
  </div>
  <!-- Library Preparation -->
  <div id="libraryPrep" class="pdf-section" style="display: none; border: 1px solid #ccc; border-radius: 8px; width: 600px; padding: 15px; box-shadow: 2px 2px 8px rgba(0,0,0,0.1); text-align: center;">
    <h3>Library Preparation</h3>
    <a href="../procedures/lib_prep.pdf" target="_blank">Download Nanopore Native Barcode Library Prep PDF</a>
    <iframe src="../procedures/lib_prep.pdf" width="100%" height="400" style="border: none; margin-top: 10px;"></iframe>
  </div>
  <!-- Flow Cell Wash and Storage -->
  <div id="flowCell" class="pdf-section" style="display: none; border: 1px solid #ccc; border-radius: 8px; width: 600px; padding: 15px; box-shadow: 2px 2px 8px rgba(0,0,0,0.1); text-align: center;">
    <h3>Flow Cell Wash and Storage</h3>
    <a href="../procedures/flow_cell_wash_kit.pdf" target="_blank">Download Flow Cell Wash and Storage Protocol PDF</a>
    <iframe src="../procedures/flow_cell_wash_kit.pdf" width="100%" height="400" style="border: none; margin-top: 10px;"></iframe>
  </div>
</div>
<!-- JavaScript for toggling sections -->
<script>
  function toggleSection(sectionId) {
    var section = document.getElementById(sectionId);
    if (section.style.display === "none" || section.style.display === "") {
      section.style.display = "block";
      section.style.opacity = 0;
      setTimeout(function() { section.style.opacity = 1; }, 10); // Fade in
    } else {
      section.style.opacity = 0;
      setTimeout(function() { section.style.display = "none"; }, 500); // Fade out
    }
  }
</script>
<!-- CSS for fade effect -->
<style>
  .pdf-section {
    transition: opacity 0.5s ease;
  }
</style>
<style>
pre code {
    background-color: #2d2d2d; /* Dark gray background for the code block */
    border: 1px solid #555555; /* Light gray border around the code block */
    border-radius: 5px; /* Rounded corners for the box */
    padding: 10px; /* Adds some space inside the box */
    display: block; /* Ensures it's displayed as a block element */
    white-space: pre-wrap; /* Ensures the code wraps properly if it's too long */
    color: #ffffff; /* White text color for better contrast */
}
</style>
<br>
<hr>
<h1 id="worked-example">Worked Example<a aria-label="Link to this section" class="paige-header-link" href="#worked-example"></a></h1>
<p>The goal of this portion of the workshop is to practice the basic bioinformatic workflow required to process nanopore adaptive sampling sequence data. Specifically, we will map reads to reference sequences, filter out host sequences, and perform homology searches against known pathogen sequences. This workflow uses publicly available data published by <a href="https://www.nature.com/articles/s41598-023-37134-9">Kipp et al. (2023)</a>.</p>
<p><a href="https://www.nature.com/articles/s41598-023-37134-9">Kipp et al. (2023)</a> shows how NAS can be used to selectively sequence DNA from multiple bacterial tick-borne pathogens circulating in wild populations of the black-legged tick vector, <i>Ixodes scapularis</i>. They test the consistency of pathogen recovery using NAS enrichment and depletion methods, compared to more traditional Illumina metagenomic sequencing methods. NAS enrichment and depletion methods successfully recovered 4 bacterial tick-borne pathogens (<i>Borellia burgdorferi</i>; <i>Borella miyamotoi</i>; <i>Anaplasma phagocytophilum</i>; and <i>Eherlichia muris</i>) from 8 female <i>I. scapularis</i> ticks, and even resulted in the sequencing of the entire bacterial pathogen genomes. Importantly, their NAS results were consistent with Illumina 16S microbiome data.</p>
<p>For this exercise, we are only working with their depletion dataset, as that workflow is best aligned with the sequencing approach of this workshop.
The goal of this process is to use NAS for real-time metagenomic surveillance, targeting pathogen detection in field-collected samples. This approach allows for selective sequencing of desired genomic regions. By performing these steps, we can efficiently map sequencing reads to reference genomes, filter out host sequences, and perform homology searches against known pathogen databases. This workflow mirrors the methodology described in <a href="https://www.nature.com/articles/s41598-023-37134-9">Kipp et al. (2023)</a>, where nanopore sequencing was used to study tick-borne bacterial pathogens.</p>
<p>Before starting with Step 1: (1) install the <a href="../programs">Required Software</a>.   (2) make sure you have downloaded Kipp&rsquo;s data as instructed in <a href="../getting-ready">Getting Ready</a>.</p>
<br>
<h2 id="step-1-getting-started">Step 1: Getting Started<a aria-label="Link to this section" class="paige-header-link" href="#step-1-getting-started"></a></h2>
<p>In this step, we are using basic Linux commands to establish our file/folder structure. Mac and Linux users may open a terminal. Windows users should open their Ubuntu terminal via WSL.</p>
<p>Print your working directory (i.e., determine where your terminal window is located within the file structure of your computer):
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="nb">pwd</span>  <span class="c1"># type this command and hit enter</span></span></span></code></pre></div></p>
<p>Home directory for Windows users via Ubuntu should be:</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="se">\\</span>wsl.localhost<span class="se">\U</span>buntu<span class="se">\h</span>ome<span class="se">\&lt;</span>USERNAME&gt;<span class="se">\f</span>ield_genomics<span class="se">\d</span>ata_example  <span class="c1"># &lt;USERNAME&gt; was defined when installing Ubuntu</span>
</span></span></code></pre></div><p>Make a directory
(folder) for this workflow:
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-python" data-lang="python"><span class="line"><span class="cl"><span class="n">mkdir</span> <span class="n">field_genomics</span></span></span></code></pre></div></p>
<p>Navigate into that directory:
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="nb">cd</span> ~/field_genomics/</span></span></code></pre></div></p>
<p>Create a new subdirectory to store example data:
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">mkdir data_example</span></span></code></pre></div></p>
<p>Create another subdirectory to store results:
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">mkdir output_example</span></span></code></pre></div></p>
<p>Create a third subdirectory to store a BLAST database:
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">mkdir blast_database</span></span></code></pre></div></p>
<p>Create a fourth subdirectory to summarize results from the example:
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">mkdir summary_example</span></span></code></pre></div></p>
<p>Copy or move the 3 data files from <a href="../getting-ready">Getting Ready</a> into the data_example/ directory.
This can be done in a File Manager or Finder window via drag-and-drop or via the command line.
Ensure that the files downloaded are named as follows, as sometimes downloading/extracting zip files from Google Drive can change file names:</p>
<ul>
<li>
<p><code>Kipp_etal_depletion_reference.fasta</code></p>
</li>
<li>
<p><code>SRR17281099.fasta.gz</code></p>
</li>
<li>
<p><code>Kipp_etal_enrichment_reference.fasta</code></p>
<p>Example code for transferring these files from your <code>Downloads/</code> folder to the <code>data_example/</code> folder:</p>
</li>
</ul>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="nb">cd</span> Downloads/
</span></span><span class="line"><span class="cl">mv SRR17281099.fasta.gz &lt;YOUR_PATH&gt;/field_genomics/
</span></span><span class="line"><span class="cl">mv Kipp_etal_* &lt;YOUR_PATH&gt;/field_genomics/
</span></span></code></pre></div><p><strong>Important</strong>: For Windows users, ensure that your computer does not suspend while running analyses. You need to repace <code>&lt;YOUR_PATH&gt;</code> with the output of your <code>pwd</code> command from above.</p>
<br>
<h2 id="step-2-burrows-wheeler-aligner-bwa">Step 2: Burrows-Wheeler Aligner (BWA)<a aria-label="Link to this section" class="paige-header-link" href="#step-2-burrows-wheeler-aligner-bwa"></a></h2>
<p>Burrows-Wheeler Alignment (BWA) is a software package that is used to align reads to a reference genome. First, we must index the reference genome. Indexing a reference genome is a process that speeds up sequence alignment by creating a data structure, allowing rapid matching of DNA sequences (reads) to the reference genome. This is how to index a reference genome:</p>
<h3 id="variables">Variables:<a aria-label="Link to this section" class="paige-header-link" href="#variables"></a></h3>
<ul>
<li><strong>Reference Genome Path</strong>: <code>data_example/Kipp_etal_depletion_reference.fasta</code></li>
</ul>
<h3 id="code">Code:<a aria-label="Link to this section" class="paige-header-link" href="#code"></a></h3>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">bwa index data_example/Kipp_etal_depletion_reference.fasta</span></span></code></pre></div>
<br>
<p><strong>Description</strong>:
<strong>What Does Indexing a Reference Genome Do?</strong></p>
<ol>
<li>
<p><strong>Input: Reference Genome</strong></p>
<ul>
<li>A FASTA file containing the reference genome sequences.</li>
<li>Example:
<pre tabindex="0"><code>&gt;chr1  
AGCTTAGCTAGCTACGATCGATCG  
&gt;chr2  
GCTAGCTAGCTAGCTGATCGTACG  
</code></pre></li>
</ul>
</li>
<li>
<p><strong>Indexing with BWA</strong></p>
<ul>
<li>BWA generates auxiliary files, such as suffix arrays or FM-indexes, that make the search process more efficient.</li>
<li>These files allow the alignment tool to quickly locate where short DNA sequences (reads) map to the genome.</li>
</ul>
</li>
<li>
<p><strong>Search Using Indexed Genome</strong></p>
<ul>
<li>When aligning reads, BWA uses the indexed reference genome to find matching sequences quickly.</li>
<li>Example:<br>
Input read: <code>AGCTAGCTAC</code><br>
Match found: <code>chr1:5-14</code></li>
</ul>
</li>
</ol>
<br>
<h2 id="step-3-align-nanopore-sequences-to-the-reference-genome">Step 3: Align Nanopore Sequences to the Reference Genome<a aria-label="Link to this section" class="paige-header-link" href="#step-3-align-nanopore-sequences-to-the-reference-genome"></a></h2>
<h3 id="variables-1">Variables:<a aria-label="Link to this section" class="paige-header-link" href="#variables-1"></a></h3>
<ul>
<li><strong>Reference Genome Path</strong>: <code>data_example/Kipp_etal_depletion_reference.fasta</code></li>
<li><strong>Nanopore Sequences Path</strong>: <code>data_example/SRR17281099.fasta.gz</code></li>
<li><strong>Output SAM File Path</strong>: <code>output_example/Kipp_etal_alig_depletion.sam</code></li>
</ul>
<h3 id="code-1">Code:<a aria-label="Link to this section" class="paige-header-link" href="#code-1"></a></h3>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">bwa mem -x ont2d data_example/Kipp_etal_depletion_reference.fasta data_example/SRR17281099.fasta.gz &gt; output_example/Kipp_etal_alig_depletion.sam</span></span></code></pre></div>
<br>
<p><strong>Description</strong>:
This step aligns the Nanopore sequencing reads to the reference genome. BWA-MEM is the algorithm in BWA that is best for aligning long reads (70bp or more).  <a href="https://samtools.github.io/hts-specs/SAMv1.pdf">SAM</a> files (the output file format) are a type of text file format that contains the alignment information of various sequences that are mapped against reference sequences. These files can also contain unmapped sequences.</p>
<div style="background-color: #f8f9fa; padding: 15px; border-radius: 8px; text-align: center;"> <img src="../procedures/mapping.png" alt="Alignment Process" style="max-width: 100%; height: auto;"/> </div>
<p>credit <a href="https://usegalaxy.org/">Galaxy</a></p>
<br>
<br>
<h2 id="step-4-filter-out-host-aligned-sequences">Step 4: Filter Out Host-Aligned Sequences<a aria-label="Link to this section" class="paige-header-link" href="#step-4-filter-out-host-aligned-sequences"></a></h2>
<p><strong>Description</strong>:
<a href="https://www.htslib.org/">samtools</a> is a suite of programs used for interacting with high-throughput sequencing data. It consists of samtools, BCFtools, and HTSlib.</p>
<h3 id="variables-2">Variables:<a aria-label="Link to this section" class="paige-header-link" href="#variables-2"></a></h3>
<ul>
<li><strong>Input SAM File Path</strong>: <code>output_example/Kipp_etal_alig_depletion.sam</code></li>
<li><strong>Output Unmapped Reads Path</strong>: <code>output_example/Kipp_etal_unmapped_depletion.fasta</code></li>
</ul>
<h3 id="code-2">Code:<a aria-label="Link to this section" class="paige-header-link" href="#code-2"></a></h3>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">samtools fasta -f <span class="m">4</span> output_example/Kipp_etal_alig_depletion.sam &gt; output_example/Kipp_etal_unmapped_depletion.fasta</span></span></code></pre></div>
<br>
<p><strong>Description</strong>:
This command filters out the mapped reads, keeping only the unmapped sequences in a FASTA file. These sequences represent reads that are NOT the host (ex. bacterial DNA, fungal DNA, contamination).</p>
<br>
<h2 id="step-5-blast">Step 5: BLAST<a aria-label="Link to this section" class="paige-header-link" href="#step-5-blast"></a></h2>
<h3 id="variables-3">Variables:<a aria-label="Link to this section" class="paige-header-link" href="#variables-3"></a></h3>
<ul>
<li><strong>BLAST Database Path</strong>: <code>blast_database/Kipp_etal_enrichment_reference.fasta</code></li>
</ul>
<h3 id="code-3">Code:<a aria-label="Link to this section" class="paige-header-link" href="#code-3"></a></h3>
<h4 id="create-blast-database">Create BLAST Database:<a aria-label="Link to this section" class="paige-header-link" href="#create-blast-database"></a></h4>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">makeblastdb -in data_example/Kipp_etal_enrichment_reference.fasta -parse_seqids -blastdb_version <span class="m">5</span> -title <span class="s2">&#34;Kipp_etal_enrichment&#34;</span> -dbtype nucl -out blast_database/Kipp_etal_enrichment_reference</span></span></code></pre></div>
<br>
<p><strong>Description</strong>:
This creates a BLAST database from the enrichment reference genome. The end goal is to identify non-host reads that match to a pathogen from a database we are creating in this step.</p>
<br>
<h4 id="run-blast-query">Run BLAST Query:<a aria-label="Link to this section" class="paige-header-link" href="#run-blast-query"></a></h4>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">blastn -query output_example/Kipp_etal_unmapped_depletion.fasta -db blast_database/Kipp_etal_enrichment_reference -out output_example/Kipp_etal_blast_results.txt</span></span></code></pre></div>
<br>
<p><strong>Description</strong>:
This step performs a BLAST search to compare the unmapped sequences against the BLAST database, producing a text file that gives us reads that matched to a pathogen databse.</p>
<br>
<p>To make it easier to summarize blast results, another BLAST query can be run, this time, defining a different type of output <code>-outfmt 6</code>. The following command outputs a tab-delimited file that allows filtering results a little easier.</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">blastn -query output_example/Kipp_etal_unmapped_depletion.fasta -db blastdb/Kipp_etal_enrichment_reference -out output_example/Kipp_etal_blast_results_table.txt -outfmt <span class="m">6</span>
</span></span><span class="line"><span class="cl">	<span class="c1">#Options supplied to -outfmt (out format) can be customized to include alignment length,      percent identity, etc.</span>
</span></span></code></pre></div><br> 
<h2 id="step-6-summarizing-result">Step 6: Summarizing result<a aria-label="Link to this section" class="paige-header-link" href="#step-6-summarizing-result"></a></h2>
<h3 id="filtering-sequence-headers-and-query-subjects-based-on-high-percentage-matches">Filtering sequence headers and query subjects based on high percentage matches<a aria-label="Link to this section" class="paige-header-link" href="#filtering-sequence-headers-and-query-subjects-based-on-high-percentage-matches"></a></h3>
<p>The following command filters the table-formated BLAST output to keep only the information for hits with a matching percentage <code>&gt;97%</code>:</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">awk -F<span class="s1">&#39;\t&#39;</span> <span class="s1">&#39;$3 &gt; 97&#39;</span> output_example/Kipp_etal_blast_results_table.txt &gt; summary_example/Kipp_etal_high_hits.txt
</span></span></code></pre></div><p>The next step is to get a list of &hellip;</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">cut -f1 summary_example/Kipp_etal_high_hits.txt <span class="p">|</span> sort <span class="p">|</span> uniq &gt; summary_example/Kipp_etal_high_hit_headers.txt
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">cut -f2 summary_example/Kipp_etal_high_hits.txt <span class="p">|</span> sort <span class="p">|</span> uniq &gt; summary_example/Kipp_etal_high_hit_subjects.txt
</span></span></code></pre></div><h3 id="separating-sequences-with-high-percentage-matching">Separating sequences with high percentage matching<a aria-label="Link to this section" class="paige-header-link" href="#separating-sequences-with-high-percentage-matching"></a></h3>
<p>Initial steps to list the headers of sequences needed:</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">zcat data_example/SRR17281099.fasta.gz &gt; data_example/SRR17281099.fasta
</span></span><span class="line"><span class="cl"> 
</span></span><span class="line"><span class="cl">samtools faidx data_example/SRR17281099.fasta
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nv">headers</span><span class="o">=</span><span class="sb">`</span>cat summary_example/Kipp_etal_high_hit_headers.txt<span class="sb">`</span>
</span></span></code></pre></div><p>Obtaining the subset of high perecentage matching sequences from the whole set of sequences.</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">samtools faidx data_example/SRR17281099.fasta <span class="nv">$headers</span> &gt; summary_example/Kipp_etal_high_hit_sequences.fasta
</span></span></code></pre></div><h3 id="exploring-results-on-ncbi-blast-online">Exploring results on NCBI-BLAST (online)<a aria-label="Link to this section" class="paige-header-link" href="#exploring-results-on-ncbi-blast-online"></a></h3>
<p>Here is a graph of the amount of hits per window. Clearly, one reference file is dominating, that being NC_021880.1_An_ph_str._JM</p>
<img src="../results/hitsperwindows.jpg" alt="Alignment Process" style="max-width: 100%; height: auto; display: block; margin: 0 auto;"/>
<p>Here is a snip of the data generated in Kipp_etal_blast_results_table.txt in Excel. It has been filtered for % Match above 97% and sorted by desencding read length. You can see that the longest high-quality hit was for the same reference as above, NC_021880.1_An_ph_str._JM.</p>
<img src="../results/results2.png" alt="Alignment Process" style="max-width: 100%; height: auto; display: block; margin: 0 auto;"/>
<p>Now, if we want to find that sequence, we can copy the Read name of the file, SRR17281099.418555, and use Ctrl + F on the Kipp_etal_high_hit_sequences.fasta to find the sequence of our read of interest:</p>
<img src="../results/finder.jpg" alt="Alignment Process" style="max-width: 100%; height: auto; display: block; margin: 0 auto;"/>
<p>Once we have this read of interest, we can use <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?PROGRAM=blastn&PAGE_TYPE=BlastSearch&LINK_LOC=blasthome">NCBI BLAST</a> to investigate of all uploaded nucleotide sequences, where does this one fall?</p>
<p>First, we paste out sequence into the nucleotide box:
<img src="../results/blast1.jpg" alt="Alignment Process" style="max-width: 100%; height: auto; display: block; margin: 0 auto;"/></p>
<p>After searching, a results pane will show you results similar to what we had on the command line, but in a different format.</p>
<img src="../results/blast2.jpg" alt="Alignment Process" style="max-width: 100%; height: auto; display: block; margin: 0 auto;"/>
<p>NCBI has a neat feature where you can view how your sequence (highlighted in yellow) compares to other uploaded sequences:</p>
<img src="../results/blast3.jpg" alt="Alignment Process" style="max-width: 100%; height: auto; display: block; margin: 0 auto;"/>
<p>In this case, we can see that our longest read maps near-perfectly to a portion of an <a href="https://www.ncbi.nlm.nih.gov/nuccore/CP000235.1">Anaplasma phagocytophilum reference genome</a>, a Gram-negative bacterium that causes multiple diseases in cattle and sheep, as well as zoonoses like <a href="https://www.ncbi.nlm.nih.gov/books/NBK513341/">human granulocytic anaplasmosis.</a></p>
<br>
<hr>
<h1 id="real-time-example">Real-Time Example<a aria-label="Link to this section" class="paige-header-link" href="#real-time-example"></a></h1>
<p><strong>Description:</strong>
This section contains all of the code we will use to analyze the Nanopore sequence data generated during the workshop. This is not part of the worked example.
<br></p>
<h2 id="1-basecalling">1. Basecalling<a aria-label="Link to this section" class="paige-header-link" href="#1-basecalling"></a></h2>
<p><strong>Description:</strong>
Basecalling is the process of determining the sequence of nucleotide bases from the raw data generated by the sequencer. For Nanopore sequencing, basecalling involves interpreting changes in electral current as single-stranded DNA (or RNA) passes through the nanopore. Current fluctuations correspond to different nucleotide bases which are decoded into <span style="color: #ff0000;">A&rsquo;s</span>,
<span style="color: #00ff00;">G&rsquo;s</span>,
<span style="color: #0000ff;">T&rsquo;s</span>,
and <span style="color: #ffa500;">C&rsquo;s</span> by the dorado basecalling software.</p>
<p><a href="../scripts/01_basecall.sh">Download 01_basecall.sh</a></p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1"># Base directory for pod5 files</span>
</span></span><span class="line"><span class="cl"><span class="nv">base_dir</span><span class="o">=</span><span class="s2">&#34;/Library/MinKNOW/data/Ecuador_Workshop_Training_Pathogens_Pero/no_sample/20240913_1820_MN40667_FAY96875_60fc85db/pod5_pass&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Output directory for basecalling results</span>
</span></span><span class="line"><span class="cl"><span class="nv">output_dir</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Ecuador_Workshop/dorado-0.7.3-linux-x64/bin/01_sup_out/&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#dorado path</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nv">dorado</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Documents/software/dorado-0.7.0-osx-arm64/bin/dorado&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Loop through each barcode directory</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="k">for</span> barcode in barcode<span class="o">{</span>01..24<span class="o">}</span> 
</span></span><span class="line"><span class="cl"><span class="c1"># this loops over files named barcode01, barcode02, ..., barcode24</span>
</span></span><span class="line"><span class="cl"><span class="c1"># The number of barcode files will depend on the number of individuals multiplexed and being analyzed simultaneously</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="k">do</span>
</span></span><span class="line"><span class="cl">    <span class="nv">$dorado</span> basecaller dna_r10.4.1_e8.2_400bps_fast@v5.0.0 <span class="s2">&#34;</span><span class="nv">$base_dir</span><span class="s2">/</span><span class="nv">$barcode</span><span class="s2">/&#34;</span> &gt; <span class="s2">&#34;</span><span class="nv">$output_dir</span><span class="s2">/</span><span class="si">${</span><span class="nv">barcode</span><span class="si">}</span><span class="s2">.bam&#34;</span>
</span></span><span class="line"><span class="cl"><span class="k">done</span></span></span></code></pre></div>
<p>Names of Dorado models (i.e., here we use <a href="https://github.com/nanoporetech/dorado?tab=readme-ov-file#dna-models">dna_r10.4.1_e8.2_400bps_fast@5.0.0</a>) are systematically structured, each segment corresponding to a different aspect of the model. Below is a breakdown of the model we used:</p>
<p><strong>Analyte Type (dna)</strong>: This denotes the type of analyte being sequenced. For DNA sequencing, it is represented as dna. If you are using a Direct RNA Sequencing Kit, this will be rna002 or rna004, depending on the kit.</p>
<p><strong>Pore Type (r10.4.1)</strong>: This section corresponds to the type of flow cell used. For instance, FLO-MIN114/FLO-FLG114 is indicated by r10.4.1, while FLO-MIN106D/FLO-FLG001 is signified by r9.4.1.</p>
<p><strong>Chemistry Type (e8.2)</strong>: This represents the chemistry type, which corresponds to the kit used for sequencing. For example, Kit 14 chemistry is denoted by e8.2 and Kit 10 or Kit 9 are denoted by e8.</p>
<p><strong>Translocation Speed (400bps)</strong>: This parameter, selected at the run setup in MinKNOW, refers to the speed of translocation. Prior to starting your run, a prompt will ask if you prefer to run at 260 bps or 400 bps. The former yields more accurate results but provides less data. As of MinKNOW version 23.04, the 260 bps option has been deprecated.</p>
<p><strong>Model Type (hac)</strong>: This represents the size of the model, where larger models yield more accurate basecalls but take more time. The three types of models are fast, hac, and sup. The fast model is the quickest, sup is the most accurate, and hac provides a balance between speed and accuracy. For most users, the hac model is recommended.</p>
<p><strong>Model Version Number (v5.0.0)</strong>: This denotes the version of the model. Model updates are regularly released, and higher version numbers typically signify greater accuracy.
For more details on decoding dorado model names see <a href="https://github.com/nanoporetech/dorado">here</a>.</p>
<br>
<h2 id="2-aligning-bam-files-to-a-reference-genome">2. Aligning BAM Files to a Reference Genome<a aria-label="Link to this section" class="paige-header-link" href="#2-aligning-bam-files-to-a-reference-genome"></a></h2>
<p>This step aligns the Nanopore sequencing reads to the reference genome. BAM files contain the same information as a SAM file, but in binary format which is not human readable. BAM files are generally smaller, however, and are therefore more efficient for software to work with. Alignment data is almost always stored as a BAM file and most software that analyzes aligned reads expects a BAM file as input.</p>
<p><a href="../scripts/02align.sh">Download 02align.sh</a></p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span>
</span></span><span class="line"><span class="cl"><span class="c1"># Path to the reference genome</span>
</span></span><span class="line"><span class="cl"><span class="nv">REFERENCE_PATH</span><span class="o">=</span><span class="s2">&#34;/Library/MinKNOW/data/References/Ecusdor_ref/pema_only_09092024.fasta&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Base directory for input BAM files </span>
</span></span><span class="line"><span class="cl"><span class="nv">INPUT_DIR</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Ecuador_Workshop/01_sup_out&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Output directory for aligned BAM files</span>
</span></span><span class="line"><span class="cl"><span class="nv">OUTPUT_DIR</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Ecuador_Workshop/02_align_out&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Loop over barcode01.bam to barcode24.bam</span>
</span></span><span class="line"><span class="cl"><span class="k">for</span> i in <span class="k">$(</span>seq -w <span class="m">1</span> 10<span class="k">)</span><span class="p">;</span> <span class="k">do</span>
</span></span><span class="line"><span class="cl">    <span class="c1"># Define input and output file names</span>
</span></span><span class="line"><span class="cl">    <span class="nv">INPUT_BAM</span><span class="o">=</span><span class="s2">&#34;</span><span class="si">${</span><span class="nv">INPUT_DIR</span><span class="si">}</span><span class="s2">/barcode</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span><span class="s2">.bam&#34;</span>
</span></span><span class="line"><span class="cl">    <span class="nv">OUTPUT_BAM</span><span class="o">=</span><span class="s2">&#34;</span><span class="si">${</span><span class="nv">OUTPUT_DIR</span><span class="si">}</span><span class="s2">/barcode</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span><span class="s2">_aligned.bam&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">    <span class="c1"># Run dorado aligner</span>
</span></span><span class="line"><span class="cl">    <span class="nb">echo</span> <span class="s2">&#34;Aligning </span><span class="si">${</span><span class="nv">INPUT_BAM</span><span class="si">}</span><span class="s2">...&#34;</span>
</span></span><span class="line"><span class="cl">    dorado aligner <span class="nv">$REFERENCE_PATH</span> <span class="nv">$INPUT_BAM</span> &gt; <span class="nv">$OUTPUT_BAM</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">    <span class="nb">echo</span> <span class="s2">&#34;Finished aligning barcode</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span><span class="s2">.bam&#34;</span>
</span></span><span class="line"><span class="cl"><span class="k">done</span></span></span></code></pre></div>
<br>
<h2 id="3-extracting-unaligned-reads">3. Extracting Unaligned Reads<a aria-label="Link to this section" class="paige-header-link" href="#3-extracting-unaligned-reads"></a></h2>
<p>Next, we extract the unaligned reads from the BAM files using <code>samtools</code>. This step is crucial for isolating reads that did not align to the reference genome, allowing us to focus on non-host sequences.</p>
<p><a href="../scripts/03_unmapped_reads.sh">Download 03_unmapped_reads.sh</a></p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span>
</span></span><span class="line"><span class="cl"><span class="c1"># Path to samtools (ensure samtools is installed and accessible in your PATH)</span>
</span></span><span class="line"><span class="cl"><span class="nv">SAMTOOLS_PATH</span><span class="o">=</span><span class="s2">&#34;/opt/homebrew/bin/samtools&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Base directory for input BAM files (update this to the correct directory)</span>
</span></span><span class="line"><span class="cl"><span class="nv">INPUT_DIR</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Ecuador_Workshop/02_align_out&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Output directory for unaligned BAM files</span>
</span></span><span class="line"><span class="cl"><span class="nv">OUTPUT_DIR</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Ecuador_Workshop/03_unmapped_out&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Loop over barcode01.bam to barcode24.bam</span>
</span></span><span class="line"><span class="cl"><span class="k">for</span> i in <span class="k">$(</span>seq -w <span class="m">1</span> 24<span class="k">)</span><span class="p">;</span> <span class="k">do</span>
</span></span><span class="line"><span class="cl">    <span class="c1"># Define input and output file names</span>
</span></span><span class="line"><span class="cl">    <span class="nv">INPUT_BAM</span><span class="o">=</span><span class="s2">&#34;</span><span class="si">${</span><span class="nv">INPUT_DIR</span><span class="si">}</span><span class="s2">/barcode</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span><span class="s2">_aligned.bam&#34;</span>
</span></span><span class="line"><span class="cl">    <span class="nv">OUTPUT_BAM</span><span class="o">=</span><span class="s2">&#34;</span><span class="si">${</span><span class="nv">OUTPUT_DIR</span><span class="si">}</span><span class="s2">/barcode</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span><span class="s2">_unaligned.bam&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">    <span class="c1"># Extract unaligned reads using samtools</span>
</span></span><span class="line"><span class="cl">    <span class="nb">echo</span> <span class="s2">&#34;Extracting unaligned reads from </span><span class="si">${</span><span class="nv">INPUT_BAM</span><span class="si">}</span><span class="s2">...&#34;</span>
</span></span><span class="line"><span class="cl">    <span class="nv">$SAMTOOLS_PATH</span> view -b -f <span class="m">4</span> <span class="nv">$INPUT_BAM</span> &gt; <span class="nv">$OUTPUT_BAM</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">    <span class="nb">echo</span> <span class="s2">&#34;Finished extracting unaligned reads for barcode</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span><span class="s2">.bam&#34;</span>
</span></span><span class="line"><span class="cl"><span class="k">done</span></span></span></code></pre></div>
<br>
<h2 id="4-renaming-sequence-ids-before-database-creation">4. Renaming Sequence IDs Before Database Creation<a aria-label="Link to this section" class="paige-header-link" href="#4-renaming-sequence-ids-before-database-creation"></a></h2>
<p>Before creating the BLAST database, we rename the sequence IDs in the FASTA file to ensure uniqueness and compatibility. This step is essential for maintaining consistency in downstream analyses.</p>
<p><a href="../scripts/03.5_shorten_fasta_header.sh">Download 03.5_shorten_fasta_header.sh</a></p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span>
</span></span><span class="line"><span class="cl">awk <span class="s1">&#39;
</span></span></span><span class="line"><span class="cl"><span class="s1">BEGIN {FS=&#34; &#34;; OFS=&#34; &#34;} 
</span></span></span><span class="line"><span class="cl"><span class="s1">/^&gt;/ {
</span></span></span><span class="line"><span class="cl"><span class="s1">    id=substr($1, 2, 40); # Get the first 40 characters of the ID
</span></span></span><span class="line"><span class="cl"><span class="s1">    count[id]++;          # Count occurrences of this ID
</span></span></span><span class="line"><span class="cl"><span class="s1">    if (count[id] &gt; 1) {  # If the ID is a duplicate, add a suffix
</span></span></span><span class="line"><span class="cl"><span class="s1">        id=id &#34;_&#34; count[id];
</span></span></span><span class="line"><span class="cl"><span class="s1">    }
</span></span></span><span class="line"><span class="cl"><span class="s1">    print &#34;&gt;&#34; id;         # Print the modified ID
</span></span></span><span class="line"><span class="cl"><span class="s1">    next;
</span></span></span><span class="line"><span class="cl"><span class="s1">}
</span></span></span><span class="line"><span class="cl"><span class="s1">{print}&#39;</span> /Users/mammalogy/Ecuador_Workshop/Pathogen_Ref/bvbrc_nsamer_good_comp_noplant_veupathbd_allbut_vecthost_pathogen_genomes_04092024.fasta &gt; /Users/mammalogy/Ecuador_Workshop/Pathogen_Ref/shortened_unique_ids.fasta</span></span></code></pre></div>
<br>
<h2 id="5-creating-a-blast-database">5. Creating a BLAST Database<a aria-label="Link to this section" class="paige-header-link" href="#5-creating-a-blast-database"></a></h2>
<p>With the sequence IDs renamed, we now create a BLAST database from the reference FASTA file. This database will be used for identifying the unaligned sequences.</p>
<p><a href="../scripts/04_blast_db_creation.sh">Download 04_blast_db_creation.sh</a></p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span>
</span></span><span class="line"><span class="cl"><span class="c1"># Make the database</span>
</span></span><span class="line"><span class="cl">makeblastdb -in /Users/mammalogy/Ecuador_Workshop/Pathogen_Ref/shortened_unique_ids.fasta -dbtype nucl -out /Users/mammalogy/Ecuador_Workshop/Pathogen_Ref/bvbrc_pathogen_db -title <span class="s2">&#34;Pathogen Genomes DB&#34;</span> -parse_seqids
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Print success message</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s2">&#34;BLAST database created successfully: </span><span class="nv">$DB_NAME</span><span class="s2">&#34;</span></span></span></code></pre></div>
<br>
<h2 id="6-converting-bam-to-fasta">6. Converting BAM to FASTA<a aria-label="Link to this section" class="paige-header-link" href="#6-converting-bam-to-fasta"></a></h2>
<p>Before running BLAST, we need to convert the unaligned BAM files into FASTA format. This step prepares the sequences for the BLAST search.</p>
<p><a href="../scripts/05_convert_bam_to_fasta.sh">Download 05_convert_bam_to_fasta.sh</a></p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span>
</span></span><span class="line"><span class="cl"><span class="c1"># Load samtools (if needed)</span>
</span></span><span class="line"><span class="cl"><span class="c1"># module load samtools</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Set paths</span>
</span></span><span class="line"><span class="cl"><span class="nv">BAM_DIR</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Ecuador_Workshop/03_unmapped_out&#34;</span>
</span></span><span class="line"><span class="cl"><span class="nv">FASTA_DIR</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Ecuador_Workshop/03_unmapped_fasta_out&#34;</span>
</span></span><span class="line"><span class="cl"><span class="nv">DB_NAME</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Ecuador_Workshop/Pathogen_Ref/bvbrc_pathogen_db&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Create output directory for FASTA files if it doesn&#39;t exist</span>
</span></span><span class="line"><span class="cl">mkdir -p <span class="nv">$FASTA_DIR</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Convert BAM files to FASTA</span>
</span></span><span class="line"><span class="cl"><span class="k">for</span> BAM_FILE in <span class="nv">$BAM_DIR</span>/*.bam<span class="p">;</span> <span class="k">do</span>
</span></span><span class="line"><span class="cl">    <span class="nv">BASENAME</span><span class="o">=</span><span class="k">$(</span>basename <span class="nv">$BAM_FILE</span> .bam<span class="k">)</span>
</span></span><span class="line"><span class="cl">    samtools fasta <span class="nv">$BAM_FILE</span> &gt; <span class="nv">$FASTA_DIR</span>/<span class="si">${</span><span class="nv">BASENAME</span><span class="si">}</span>.fasta
</span></span><span class="line"><span class="cl">    <span class="nb">echo</span> <span class="s2">&#34;Converted </span><span class="nv">$BAM_FILE</span><span class="s2"> to FASTA format&#34;</span>
</span></span><span class="line"><span class="cl"><span class="k">done</span></span></span></code></pre></div>
<br>
<h2 id="7-running-blast-on-fasta-files">7. Running BLAST on FASTA Files<a aria-label="Link to this section" class="paige-header-link" href="#7-running-blast-on-fasta-files"></a></h2>
<p>The converted FASTA files are then subjected to BLAST search against the created database to identify possible pathogenic sequences.</p>
<p><a href="../scripts/06_blast_fastas.sh">Download 06_blast_fastas.sh</a></p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span>
</span></span><span class="line"><span class="cl"><span class="c1"># Set paths</span>
</span></span><span class="line"><span class="cl"><span class="nv">FASTA_DIR</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Ecuador_Workshop/03_unmapped_fasta_out&#34;</span>
</span></span><span class="line"><span class="cl"><span class="nv">DB_NAME</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Ecuador_Workshop/Pathogen_Ref/bvbrc_pathogen_db&#34;</span>
</span></span><span class="line"><span class="cl"><span class="nv">OUTPUT_DIR</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Ecuador_Workshop/03_blastn_results&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Create output directory for BLAST results if it doesn&#39;t exist</span>
</span></span><span class="line"><span class="cl">mkdir -p <span class="nv">$OUTPUT_DIR</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Run BLAST for each FASTA file</span>
</span></span><span class="line"><span class="cl"><span class="k">for</span> FASTA_FILE in <span class="nv">$FASTA_DIR</span>/*.fasta<span class="p">;</span> <span class="k">do</span>
</span></span><span class="line"><span class="cl">    <span class="nv">BASENAME</span><span class="o">=</span><span class="k">$(</span>basename <span class="nv">$FASTA_FILE</span> .fasta<span class="k">)</span>
</span></span><span class="line"><span class="cl">    blastn -query <span class="nv">$FASTA_FILE</span> -db <span class="nv">$DB_NAME</span> -out <span class="nv">$OUTPUT_DIR</span>/<span class="si">${</span><span class="nv">BASENAME</span><span class="si">}</span>_blastn.txt -outfmt <span class="m">6</span> -num_threads <span class="m">16</span>
</span></span><span class="line"><span class="cl">    <span class="nb">echo</span> <span class="s2">&#34;BLAST completed for </span><span class="nv">$FASTA_FILE</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl"><span class="k">done</span></span></span></code></pre></div>
<br>
<h2 id="8-renaming-sequence-ids-in-blast-results">8. Renaming Sequence IDs in BLAST Results<a aria-label="Link to this section" class="paige-header-link" href="#8-renaming-sequence-ids-in-blast-results"></a></h2>
<p>Finally, the sequence IDs in the BLAST output are renamed to match their original identifiers, providing clear traceability of the sequences.</p>
<p><a href="../scripts/07_rename_seqIDs.sh">Download 07_rename_seqIDs.sh</a></p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span>
</span></span><span class="line"><span class="cl"><span class="c1"># Define directories and files</span>
</span></span><span class="line"><span class="cl"><span class="nv">RESULTS_DIR</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Ecuador_Workshop/03_blastn_results&#34;</span>
</span></span><span class="line"><span class="cl"><span class="nv">MAPPING_FILE</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Ecuador_Workshop/seq_id_mapping.txt&#34;</span>
</span></span><span class="line"><span class="cl"><span class="nv">OUTPUT_DIR</span><span class="o">=</span><span class="s2">&#34;/Users/mammalogy/Ecuador_Workshop/03_blastn_results_with_original_ids&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Create output directory if it doesn&#39;t exist</span>
</span></span><span class="line"><span class="cl">mkdir -p <span class="nv">$OUTPUT_DIR</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Loop through each BLAST result file from barcode01 to barcode24</span>
</span></span><span class="line"><span class="cl"><span class="k">for</span> i in <span class="k">$(</span>seq -f <span class="s2">&#34;%02g&#34;</span> <span class="m">1</span> 24<span class="k">)</span><span class="p">;</span> <span class="k">do</span>
</span></span><span class="line"><span class="cl">    <span class="nv">INPUT_FILE</span><span class="o">=</span><span class="s2">&#34;</span><span class="nv">$RESULTS_DIR</span><span class="s2">/barcode</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span><span class="s2">_unaligned_blastn.txt&#34;</span>
</span></span><span class="line"><span class="cl">    <span class="nv">OUTPUT_FILE</span><span class="o">=</span><span class="s2">&#34;</span><span class="nv">$OUTPUT_DIR</span><span class="s2">/barcode</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span><span class="s2">_unaligned_blastn_with_original_ids.txt&#34;</span>
</span></span><span class="line"><span class="cl">    
</span></span><span class="line"><span class="cl">    <span class="c1"># Replace seq_X IDs with original IDs using awk</span>
</span></span><span class="line"><span class="cl">    awk <span class="s1">&#39;NR==FNR {map[$1]=$2; next} $2 in map {$2=map[$2]}1&#39;</span> <span class="nv">$MAPPING_FILE</span> <span class="nv">$INPUT_FILE</span> &gt; <span class="nv">$OUTPUT_FILE</span>
</span></span><span class="line"><span class="cl">    
</span></span><span class="line"><span class="cl">    <span class="nb">echo</span> <span class="s2">&#34;Processed </span><span class="nv">$INPUT_FILE</span><span class="s2"> -&gt; </span><span class="nv">$OUTPUT_FILE</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl"><span class="k">done</span></span></span></code></pre></div>
</main>
        

        









    

    




    </div>
</article>






























    
    



    



    



    



    



    



    



    



    



    






                    











    





    <footer id="paige-site-footer">
        

        
            <p class="paige-row-narrow text-center text-secondary" id="paige-copyright">Â© Alex Hey</p>
        

        

        

        
    </footer>



                    
                </div>
            </div>
        </div>

        















    
    
    


<script  crossorigin="anonymous"   defer  integrity="sha256-gfPiYYSP&#43;RNtNeRBt45mJombFJmNf0V0Ipu785Obi0M="   referrerpolicy="no-referrer"  src="../js/paige/bootstrap/bootstrap.bundle.min.81f3e261848ff9136d35e441b78e6626899b14998d7f4574229bbbf3939b8b43.js" ></script>







    



    



    



    



    



    



    



    



    



    



    



    



    



    



    





<noscript>JavaScript is required</noscript>


        
    </body>
</html>
